<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.5"/>
<title>Proyecto JGAP: Referencia de la Clase org.jgap.Genotype</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="navtree.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="resize.js"></script>
<script type="text/javascript" src="navtree.js"></script>
<script type="text/javascript">
  $(document).ready(initResizable);
  $(window).load(resizeHeight);
</script>
<link href="search/search.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="search/search.js"></script>
<script type="text/javascript">
  $(document).ready(function() { searchBox.OnSelectItem(0); });
</script>
<link href="doxygen.css" rel="stylesheet" type="text/css" />
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<div id="titlearea">
<table cellspacing="0" cellpadding="0">
 <tbody>
 <tr style="height: 56px;">
  <td style="padding-left: 0.5em;">
   <div id="projectname">Proyecto JGAP
   &#160;<span id="projectnumber">1.0</span>
   </div>
   <div id="projectbrief">Proyecto JGAP</div>
  </td>
 </tr>
 </tbody>
</table>
</div>
<!-- end header part -->
<!-- Generado por Doxygen 1.8.5 -->
<script type="text/javascript">
var searchBox = new SearchBox("searchBox", "search",false,'Buscar');
</script>
  <div id="navrow1" class="tabs">
    <ul class="tablist">
      <li><a href="index.html"><span>Página&#160;principal</span></a></li>
      <li><a href="pages.html"><span>Páginas&#160;relacionadas</span></a></li>
      <li><a href="namespaces.html"><span>Paquetes</span></a></li>
      <li class="current"><a href="annotated.html"><span>Clases</span></a></li>
      <li><a href="files.html"><span>Archivos</span></a></li>
      <li>
        <div id="MSearchBox" class="MSearchBoxInactive">
        <span class="left">
          <img id="MSearchSelect" src="search/mag_sel.png"
               onmouseover="return searchBox.OnSearchSelectShow()"
               onmouseout="return searchBox.OnSearchSelectHide()"
               alt=""/>
          <input type="text" id="MSearchField" value="Buscar" accesskey="S"
               onfocus="searchBox.OnSearchFieldFocus(true)" 
               onblur="searchBox.OnSearchFieldFocus(false)" 
               onkeyup="searchBox.OnSearchFieldChange(event)"/>
          </span><span class="right">
            <a id="MSearchClose" href="javascript:searchBox.CloseResultsWindow()"><img id="MSearchCloseImg" border="0" src="search/close.png" alt=""/></a>
          </span>
        </div>
      </li>
    </ul>
  </div>
  <div id="navrow2" class="tabs2">
    <ul class="tablist">
      <li><a href="annotated.html"><span>Lista&#160;de&#160;clases</span></a></li>
      <li><a href="classes.html"><span>Índice&#160;de&#160;clases</span></a></li>
      <li><a href="hierarchy.html"><span>Jerarquía&#160;de&#160;la&#160;clase</span></a></li>
      <li><a href="functions.html"><span>Miembros&#160;de&#160;las&#160;clases</span></a></li>
    </ul>
  </div>
</div><!-- top -->
<div id="side-nav" class="ui-resizable side-nav-resizable">
  <div id="nav-tree">
    <div id="nav-tree-contents">
      <div id="nav-sync" class="sync"></div>
    </div>
  </div>
  <div id="splitbar" style="-moz-user-select:none;" 
       class="ui-resizable-handle">
  </div>
</div>
<script type="text/javascript">
$(document).ready(function(){initNavTree('classorg_1_1jgap_1_1_genotype.html','');});
</script>
<div id="doc-content">
<!-- window showing the filter options -->
<div id="MSearchSelectWindow"
     onmouseover="return searchBox.OnSearchSelectShow()"
     onmouseout="return searchBox.OnSearchSelectHide()"
     onkeydown="return searchBox.OnSearchSelectKey(event)">
<a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(0)"><span class="SelectionMark">&#160;</span>Todo</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(1)"><span class="SelectionMark">&#160;</span>Clases</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(2)"><span class="SelectionMark">&#160;</span>Namespaces</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(3)"><span class="SelectionMark">&#160;</span>Archivos</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(4)"><span class="SelectionMark">&#160;</span>Funciones</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(5)"><span class="SelectionMark">&#160;</span>Variables</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(6)"><span class="SelectionMark">&#160;</span>Páginas</a></div>

<!-- iframe showing the search results (closed by default) -->
<div id="MSearchResultsWindow">
<iframe src="javascript:void(0)" frameborder="0" 
        name="MSearchResults" id="MSearchResults">
</iframe>
</div>

<div class="header">
  <div class="summary">
<a href="#pub-methods">Métodos públicos</a> &#124;
<a href="#pub-static-methods">Métodos públicos estáticos</a> &#124;
<a href="#pro-methods">Métodos protegidos</a> &#124;
<a href="classorg_1_1jgap_1_1_genotype-members.html">Lista de todos los miembros</a>  </div>
  <div class="headertitle">
<div class="title">Referencia de la Clase org.jgap.Genotype</div>  </div>
</div><!--header-->
<div class="contents">

<p>Herencias Serializable y Runnable.</p>

<p>Heredado por <a class="el" href="classorg_1_1jgap_1_1distr_1_1_distributed_genotype.html">org.jgap.distr.DistributedGenotype</a>.</p>
<table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pub-methods"></a>
Métodos públicos</h2></td></tr>
<tr class="memitem:a00b11af9a68e9d84bbc52afbf661607f"><td class="memItemLeft" align="right" valign="top">&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype.html#a00b11af9a68e9d84bbc52afbf661607f">Genotype</a> (<a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a> a_configuration, <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html">IChromosome</a>[] a_initialChromosomes)  throws InvalidConfigurationException </td></tr>
<tr class="separator:a00b11af9a68e9d84bbc52afbf661607f"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a7b08ffd2154cd3ccc8636b795666c996"><td class="memItemLeft" align="right" valign="top">&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype.html#a7b08ffd2154cd3ccc8636b795666c996">Genotype</a> (<a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a> a_configuration, <a class="el" href="classorg_1_1jgap_1_1_population.html">Population</a> a_population)  throws InvalidConfigurationException </td></tr>
<tr class="separator:a7b08ffd2154cd3ccc8636b795666c996"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:afc52d6b649eea1eefd92e376ea3762ff"><td class="memItemLeft" align="right" valign="top">&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype.html#afc52d6b649eea1eefd92e376ea3762ff">Genotype</a> (<a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a> a_configuration)  throws InvalidConfigurationException </td></tr>
<tr class="separator:afc52d6b649eea1eefd92e376ea3762ff"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ad6795433da5538c6f520253ff9d2382e"><td class="memItemLeft" align="right" valign="top">synchronized <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html">IChromosome</a>[]&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype.html#ad6795433da5538c6f520253ff9d2382e">getChromosomes</a> ()</td></tr>
<tr class="separator:ad6795433da5538c6f520253ff9d2382e"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a8241b2d89d32b283abe5c9b42974d64d"><td class="memItemLeft" align="right" valign="top"><a class="el" href="classorg_1_1jgap_1_1_population.html">Population</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype.html#a8241b2d89d32b283abe5c9b42974d64d">getPopulation</a> ()</td></tr>
<tr class="separator:a8241b2d89d32b283abe5c9b42974d64d"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a05380af4c0737841c09d3a5533f9f942"><td class="memItemLeft" align="right" valign="top">synchronized <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html">IChromosome</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype.html#a05380af4c0737841c09d3a5533f9f942">getFittestChromosome</a> ()</td></tr>
<tr class="separator:a05380af4c0737841c09d3a5533f9f942"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a6c40835533352287db8d34e5fae6875f"><td class="memItemLeft" align="right" valign="top">synchronized <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html">IChromosome</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype.html#a6c40835533352287db8d34e5fae6875f">getFittestChromosome</a> (int a_startIndex, int a_endIndex)</td></tr>
<tr class="separator:a6c40835533352287db8d34e5fae6875f"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a602d6b1413007ef28880cf88ccd2f912"><td class="memItemLeft" align="right" valign="top">synchronized List&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype.html#a602d6b1413007ef28880cf88ccd2f912">getFittestChromosomes</a> (int a_numberOfChromosomes)</td></tr>
<tr class="separator:a602d6b1413007ef28880cf88ccd2f912"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a54a6791fbb6e13920e79f4e2dd719eb4"><td class="memItemLeft" align="right" valign="top">synchronized void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype.html#a54a6791fbb6e13920e79f4e2dd719eb4">evolve</a> ()</td></tr>
<tr class="separator:a54a6791fbb6e13920e79f4e2dd719eb4"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ab15d88cf1a6084f9efea2dc935968471"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype.html#ab15d88cf1a6084f9efea2dc935968471">evolve</a> (int a_numberOfEvolutions)</td></tr>
<tr class="separator:ab15d88cf1a6084f9efea2dc935968471"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aa740dd753d31620166979199a38d5a50"><td class="memItemLeft" align="right" valign="top">List&lt; String &gt;&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype.html#aa740dd753d31620166979199a38d5a50">evolve</a> (IEvolutionMonitor a_monitor)</td></tr>
<tr class="separator:aa740dd753d31620166979199a38d5a50"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a658a541940e9be8ef474375091ff31ae"><td class="memItemLeft" align="right" valign="top">String&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype.html#a658a541940e9be8ef474375091ff31ae">toString</a> ()</td></tr>
<tr class="separator:a658a541940e9be8ef474375091ff31ae"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ac622bbf487c0ab7344d46ca669607cef"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype.html#ac622bbf487c0ab7344d46ca669607cef">fillPopulation</a> (final int a_num)  throws InvalidConfigurationException </td></tr>
<tr class="separator:ac622bbf487c0ab7344d46ca669607cef"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ab9bba698920b358bb53131338c9d130a"><td class="memItemLeft" align="right" valign="top">boolean&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype.html#ab9bba698920b358bb53131338c9d130a">equals</a> (Object a_other)</td></tr>
<tr class="separator:ab9bba698920b358bb53131338c9d130a"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:adf2dbe6d60d6c74592cb1e6398cece2c"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype.html#adf2dbe6d60d6c74592cb1e6398cece2c">applyGeneticOperators</a> ()</td></tr>
<tr class="separator:adf2dbe6d60d6c74592cb1e6398cece2c"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a786848d2da4fd3d87f9f4caa8f9dd58b"><td class="memItemLeft" align="right" valign="top"><a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype.html#a786848d2da4fd3d87f9f4caa8f9dd58b">getConfiguration</a> ()</td></tr>
<tr class="separator:a786848d2da4fd3d87f9f4caa8f9dd58b"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:abe044a5473e935c2093ee3c6c18775f7"><td class="memItemLeft" align="right" valign="top">int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype.html#abe044a5473e935c2093ee3c6c18775f7">hashCode</a> ()</td></tr>
<tr class="separator:abe044a5473e935c2093ee3c6c18775f7"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a0d14c5400197e3c43f9d1f0666309a85"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype.html#a0d14c5400197e3c43f9d1f0666309a85">run</a> ()</td></tr>
<tr class="separator:a0d14c5400197e3c43f9d1f0666309a85"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a84fd1b399188ea10c0af4b7c1152e0c2"><td class="memItemLeft" align="right" valign="top">List&lt; IEvolveJob &gt;&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype.html#a84fd1b399188ea10c0af4b7c1152e0c2">getEvolves</a> (IPopulationSplitter a_splitter)  throws Exception </td></tr>
<tr class="separator:a84fd1b399188ea10c0af4b7c1152e0c2"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aa2b0341edc420ad61e797508b8bf4dd5"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype.html#aa2b0341edc420ad61e797508b8bf4dd5">mergeResults</a> (IPopulationMerger a_merger, EvolveResult[] a_results)  throws Exception </td></tr>
<tr class="separator:aa2b0341edc420ad61e797508b8bf4dd5"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:abba6c68b3e8388be6884a35f5600e8cb"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype.html#abba6c68b3e8388be6884a35f5600e8cb">setUseMonitor</a> (boolean a_useMonitor)</td></tr>
<tr class="separator:abba6c68b3e8388be6884a35f5600e8cb"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:af3f49ba90cc8515817366792b7bce320"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype.html#af3f49ba90cc8515817366792b7bce320">setMonitor</a> (IEvolutionMonitor a_monitor)</td></tr>
<tr class="separator:af3f49ba90cc8515817366792b7bce320"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pub-static-methods"></a>
Métodos públicos estáticos</h2></td></tr>
<tr class="memitem:a930184e2920b4642f20d0b8bc8412005"><td class="memItemLeft" align="right" valign="top">static <a class="el" href="classorg_1_1jgap_1_1_genotype.html">Genotype</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype.html#a930184e2920b4642f20d0b8bc8412005">randomInitialGenotype</a> (<a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a> a_configuration)  throws InvalidConfigurationException </td></tr>
<tr class="separator:a930184e2920b4642f20d0b8bc8412005"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aaf3d4bf0b470aa4655e85262289443eb"><td class="memItemLeft" align="right" valign="top">static <a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype.html#aaf3d4bf0b470aa4655e85262289443eb">getStaticConfiguration</a> ()</td></tr>
<tr class="separator:aaf3d4bf0b470aa4655e85262289443eb"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a2f44a7709eca83cddd6d794ee59e663c"><td class="memItemLeft" align="right" valign="top">static void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype.html#a2f44a7709eca83cddd6d794ee59e663c">setStaticConfiguration</a> (<a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a> a_configuration)</td></tr>
<tr class="separator:a2f44a7709eca83cddd6d794ee59e663c"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pro-methods"></a>
Métodos protegidos</h2></td></tr>
<tr class="memitem:a6c14c3a36aa8e4291840f0b9c0facb35"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype.html#a6c14c3a36aa8e4291840f0b9c0facb35">applyNaturalSelectors</a> (boolean a_processBeforeGeneticOperators)</td></tr>
<tr class="separator:a6c14c3a36aa8e4291840f0b9c0facb35"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ac56a4d112334db07ba333602c4a24706"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype.html#ac56a4d112334db07ba333602c4a24706">setPopulation</a> (<a class="el" href="classorg_1_1jgap_1_1_population.html">Population</a> a_pop)</td></tr>
<tr class="separator:ac56a4d112334db07ba333602c4a24706"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a554d1eaf21a0a9437cdbc5b0fbeedef4"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype.html#a554d1eaf21a0a9437cdbc5b0fbeedef4">applyGeneticOperator</a> (<a class="el" href="interfaceorg_1_1jgap_1_1_genetic_operator.html">GeneticOperator</a> a_operator, <a class="el" href="classorg_1_1jgap_1_1_population.html">Population</a> a_population, List a_chromosomes)</td></tr>
<tr class="separator:a554d1eaf21a0a9437cdbc5b0fbeedef4"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:af5e224c5b7c05dbe02b3324d5ec16f2a"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classorg_1_1jgap_1_1_genotype.html#af5e224c5b7c05dbe02b3324d5ec16f2a">keepPopSizeConstant</a> (<a class="el" href="classorg_1_1jgap_1_1_population.html">Population</a> a_pop, int a_maxSize)</td></tr>
<tr class="separator:af5e224c5b7c05dbe02b3324d5ec16f2a"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table>
<a name="details" id="details"></a><h2 class="groupheader">Descripción detallada</h2>
<div class="textblock"><p>Genotypes are fixed-length populations of chromosomes. As an instance of a <a class="el" href="classorg_1_1jgap_1_1_genotype.html">Genotype</a> is evolved, all of its Chromosomes are also evolved. A <a class="el" href="classorg_1_1jgap_1_1_genotype.html">Genotype</a> may be constructed normally via constructor, or the static <a class="el" href="classorg_1_1jgap_1_1_genotype.html#a930184e2920b4642f20d0b8bc8412005">randomInitialGenotype()</a> method can be used to generate a <a class="el" href="classorg_1_1jgap_1_1_genotype.html">Genotype</a> with a randomized <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> population. </p>
<p>Please note that among all created <a class="el" href="classorg_1_1jgap_1_1_genotype.html">Genotype</a> instances there may only be one configuration, used by all <a class="el" href="classorg_1_1jgap_1_1_genotype.html">Genotype</a> instances.</p>
<dl class="section author"><dt>Autor</dt><dd>Neil Rotstan </dd>
<dd>
Klaus Meffert </dd></dl>
<dl class="section since"><dt>Desde</dt><dd>1.0 </dd></dl>
</div><h2 class="groupheader">Documentación del constructor y destructor</h2>
<a class="anchor" id="a00b11af9a68e9d84bbc52afbf661607f"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">org.jgap.Genotype.Genotype </td>
          <td>(</td>
          <td class="paramtype"><a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a>&#160;</td>
          <td class="paramname"><em>a_configuration</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype"><a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html">IChromosome</a>[]&#160;</td>
          <td class="paramname"><em>a_initialChromosomes</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td> throws <a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a></td>
        </tr>
      </table>
</div><div class="memdoc">
<p>Constructs a new <a class="el" href="classorg_1_1jgap_1_1_genotype.html">Genotype</a> instance with the given array of Chromosomes and the given active <a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a> instance. Note that the <a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a> object must be in a valid state when this method is invoked, or a <a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a> will be thrown.</p>
<dl class="params"><dt>Parámetros</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_configuration</td><td>the <a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a> object to use </td></tr>
    <tr><td class="paramname">a_initialChromosomes</td><td>the <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> population to be managed by this <a class="el" href="classorg_1_1jgap_1_1_genotype.html">Genotype</a> instance </td></tr>
  </table>
  </dd>
</dl>
<dl class="exception"><dt>Excepciones</dt><dd>
  <table class="exception">
    <tr><td class="paramname"><a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a></td><td>if the given <a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a> object is in an invalid state</td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Autor</dt><dd>Neil Rotstan </dd>
<dd>
Klaus Meffert </dd></dl>
<dl class="section since"><dt>Desde</dt><dd>1.0 </dd></dl>
<dl class="deprecated"><dt><b><a class="el" href="deprecated.html#_deprecated000007">Obsoleto:</a></b></dt><dd>use <a class="el" href="classorg_1_1jgap_1_1_genotype.html#a7b08ffd2154cd3ccc8636b795666c996">Genotype(Configuration, Population)</a> instead </dd></dl>
<div class="fragment"><div class="line"><a name="l00078"></a><span class="lineno">   78</span>&#160;                                           {</div>
<div class="line"><a name="l00079"></a><span class="lineno">   79</span>&#160;    <span class="keyword">this</span>(a_configuration, <span class="keyword">new</span> Population(a_configuration, a_initialChromosomes));</div>
<div class="line"><a name="l00080"></a><span class="lineno">   80</span>&#160;  }</div>
</div><!-- fragment -->
</div>
</div>
<a class="anchor" id="a7b08ffd2154cd3ccc8636b795666c996"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">org.jgap.Genotype.Genotype </td>
          <td>(</td>
          <td class="paramtype"><a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a>&#160;</td>
          <td class="paramname"><em>a_configuration</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype"><a class="el" href="classorg_1_1jgap_1_1_population.html">Population</a>&#160;</td>
          <td class="paramname"><em>a_population</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td> throws <a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a></td>
        </tr>
      </table>
</div><div class="memdoc">
<p>Constructs a new <a class="el" href="classorg_1_1jgap_1_1_genotype.html">Genotype</a> instance with the given array of Chromosomes and the given active <a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a> instance. Note that the <a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a> object must be in a valid state when this method is invoked, or a InvalidconfigurationException will be thrown.</p>
<dl class="params"><dt>Parámetros</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_configuration</td><td>the <a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a> object to use </td></tr>
    <tr><td class="paramname">a_population</td><td>the <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> population to be managed by this <a class="el" href="classorg_1_1jgap_1_1_genotype.html">Genotype</a> instance </td></tr>
  </table>
  </dd>
</dl>
<dl class="exception"><dt>Excepciones</dt><dd>
  <table class="exception">
    <tr><td class="paramname"><a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a></td><td></td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Autor</dt><dd>Neil Rotstan </dd>
<dd>
Klaus Meffert </dd></dl>
<dl class="section since"><dt>Desde</dt><dd>2.0 </dd></dl>
<div class="fragment"><div class="line"><a name="l00099"></a><span class="lineno">   99</span>&#160;                                           {</div>
<div class="line"><a name="l00100"></a><span class="lineno">  100</span>&#160;    <span class="comment">// Sanity checks: Make sure neither the Configuration, nor the array</span></div>
<div class="line"><a name="l00101"></a><span class="lineno">  101</span>&#160;    <span class="comment">// of Chromosomes, nor any of the Genes inside the array is null.</span></div>
<div class="line"><a name="l00102"></a><span class="lineno">  102</span>&#160;    <span class="comment">// -----------------------------------------------------------------</span></div>
<div class="line"><a name="l00103"></a><span class="lineno">  103</span>&#160;    <span class="keywordflow">if</span> (a_configuration == null) {</div>
<div class="line"><a name="l00104"></a><span class="lineno">  104</span>&#160;      <span class="keywordflow">throw</span> <span class="keyword">new</span> IllegalArgumentException(</div>
<div class="line"><a name="l00105"></a><span class="lineno">  105</span>&#160;          <span class="stringliteral">&quot;The Configuration instance must not be null.&quot;</span>);</div>
<div class="line"><a name="l00106"></a><span class="lineno">  106</span>&#160;    }</div>
<div class="line"><a name="l00107"></a><span class="lineno">  107</span>&#160;    <span class="keywordflow">if</span> (a_population == null) {</div>
<div class="line"><a name="l00108"></a><span class="lineno">  108</span>&#160;      <span class="keywordflow">throw</span> <span class="keyword">new</span> IllegalArgumentException(</div>
<div class="line"><a name="l00109"></a><span class="lineno">  109</span>&#160;          <span class="stringliteral">&quot;The Population must not be null.&quot;</span>);</div>
<div class="line"><a name="l00110"></a><span class="lineno">  110</span>&#160;    }</div>
<div class="line"><a name="l00111"></a><span class="lineno">  111</span>&#160;    <span class="keywordflow">for</span> (<span class="keywordtype">int</span> i = 0; i &lt; a_population.size(); i++) {</div>
<div class="line"><a name="l00112"></a><span class="lineno">  112</span>&#160;      <span class="keywordflow">if</span> (a_population.getChromosome(i) == null) {</div>
<div class="line"><a name="l00113"></a><span class="lineno">  113</span>&#160;        <span class="keywordflow">throw</span> <span class="keyword">new</span> IllegalArgumentException(</div>
<div class="line"><a name="l00114"></a><span class="lineno">  114</span>&#160;            <span class="stringliteral">&quot;The Chromosome instance at index &quot;</span> + i + <span class="stringliteral">&quot; of the array of &quot;</span> +</div>
<div class="line"><a name="l00115"></a><span class="lineno">  115</span>&#160;            <span class="stringliteral">&quot;Chromosomes is null. No Chromosomes instance in this array &quot;</span> +</div>
<div class="line"><a name="l00116"></a><span class="lineno">  116</span>&#160;            <span class="stringliteral">&quot;must not be null.&quot;</span>);</div>
<div class="line"><a name="l00117"></a><span class="lineno">  117</span>&#160;      }</div>
<div class="line"><a name="l00118"></a><span class="lineno">  118</span>&#160;    }</div>
<div class="line"><a name="l00119"></a><span class="lineno">  119</span>&#160;    m_population = a_population;</div>
<div class="line"><a name="l00120"></a><span class="lineno">  120</span>&#160;    <span class="comment">// Lock the settings of the configuration object so that it cannot</span></div>
<div class="line"><a name="l00121"></a><span class="lineno">  121</span>&#160;    <span class="comment">// be altered.</span></div>
<div class="line"><a name="l00122"></a><span class="lineno">  122</span>&#160;    <span class="comment">// ---------------------------------------------------------------</span></div>
<div class="line"><a name="l00123"></a><span class="lineno">  123</span>&#160;    a_configuration.lockSettings();</div>
<div class="line"><a name="l00124"></a><span class="lineno">  124</span>&#160;    m_activeConfiguration = a_configuration;</div>
<div class="line"><a name="l00125"></a><span class="lineno">  125</span>&#160;  }</div>
</div><!-- fragment -->
</div>
</div>
<a class="anchor" id="afc52d6b649eea1eefd92e376ea3762ff"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">org.jgap.Genotype.Genotype </td>
          <td>(</td>
          <td class="paramtype"><a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a>&#160;</td>
          <td class="paramname"><em>a_configuration</em></td><td>)</td>
          <td> throws <a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a></td>
        </tr>
      </table>
</div><div class="memdoc">
<p>Don't use this constructor, it's only for internal use.</p>
<dl class="params"><dt>Parámetros</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_configuration</td><td>not used here! </td></tr>
  </table>
  </dd>
</dl>
<dl class="exception"><dt>Excepciones</dt><dd>
  <table class="exception">
    <tr><td class="paramname"><a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a></td><td></td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Autor</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Desde</dt><dd>3.0 </dd></dl>
<div class="fragment"><div class="line"><a name="l00137"></a><span class="lineno">  137</span>&#160;                                           {</div>
<div class="line"><a name="l00138"></a><span class="lineno">  138</span>&#160;  }</div>
</div><!-- fragment -->
</div>
</div>
<h2 class="groupheader">Documentación de las funciones miembro</h2>
<a class="anchor" id="a554d1eaf21a0a9437cdbc5b0fbeedef4"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.Genotype.applyGeneticOperator </td>
          <td>(</td>
          <td class="paramtype"><a class="el" href="interfaceorg_1_1jgap_1_1_genetic_operator.html">GeneticOperator</a>&#160;</td>
          <td class="paramname"><em>a_operator</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype"><a class="el" href="classorg_1_1jgap_1_1_population.html">Population</a>&#160;</td>
          <td class="paramname"><em>a_population</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">List&#160;</td>
          <td class="paramname"><em>a_chromosomes</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">protected</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Overwritable method that calls a <a class="el" href="interfaceorg_1_1jgap_1_1_genetic_operator.html">GeneticOperator</a> to operate on a given population and asks him to store the result in the list of chromosomes. Override this method if you want to ensure that a_chromosomes is not part of a_population resp. if you want to use a different list.</p>
<dl class="params"><dt>Parámetros</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_operator</td><td>the <a class="el" href="interfaceorg_1_1jgap_1_1_genetic_operator.html">GeneticOperator</a> to call </td></tr>
    <tr><td class="paramname">a_population</td><td>the population to use </td></tr>
    <tr><td class="paramname">a_chromosomes</td><td>the list of <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> objects to return</td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Autor</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Desde</dt><dd>2.4 </dd></dl>
<div class="fragment"><div class="line"><a name="l00589"></a><span class="lineno">  589</span>&#160;                                                          {</div>
<div class="line"><a name="l00590"></a><span class="lineno">  590</span>&#160;    a_operator.operate(a_population, a_chromosomes);</div>
<div class="line"><a name="l00591"></a><span class="lineno">  591</span>&#160;  }</div>
</div><!-- fragment -->
</div>
</div>
<a class="anchor" id="adf2dbe6d60d6c74592cb1e6398cece2c"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.Genotype.applyGeneticOperators </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
</div><div class="memdoc">
<p>Applies all GeneticOperators registered with the <a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a>.</p>
<dl class="section author"><dt>Autor</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Desde</dt><dd>3.0 </dd></dl>
<div class="fragment"><div class="line"><a name="l00501"></a><span class="lineno">  501</span>&#160;                                      {</div>
<div class="line"><a name="l00502"></a><span class="lineno">  502</span>&#160;    List geneticOperators = m_activeConfiguration.getGeneticOperators();</div>
<div class="line"><a name="l00503"></a><span class="lineno">  503</span>&#160;    Iterator operatorIterator = geneticOperators.iterator();</div>
<div class="line"><a name="l00504"></a><span class="lineno">  504</span>&#160;    <span class="keywordflow">while</span> (operatorIterator.hasNext()) {</div>
<div class="line"><a name="l00505"></a><span class="lineno">  505</span>&#160;      GeneticOperator operator = (GeneticOperator) operatorIterator.next();</div>
<div class="line"><a name="l00506"></a><span class="lineno">  506</span>&#160;      <a class="code" href="classorg_1_1jgap_1_1_genotype.html#a554d1eaf21a0a9437cdbc5b0fbeedef4">applyGeneticOperator</a>(operator, <a class="code" href="classorg_1_1jgap_1_1_genotype.html#a8241b2d89d32b283abe5c9b42974d64d">getPopulation</a>(),</div>
<div class="line"><a name="l00507"></a><span class="lineno">  507</span>&#160;                           getPopulation().<a class="code" href="classorg_1_1jgap_1_1_genotype.html#ad6795433da5538c6f520253ff9d2382e">getChromosomes</a>());</div>
<div class="line"><a name="l00508"></a><span class="lineno">  508</span>&#160;<span class="comment">//      List workingPool = new Vector();</span></div>
<div class="line"><a name="l00509"></a><span class="lineno">  509</span>&#160;<span class="comment">//      applyGeneticOperator(operator, getPopulation(),</span></div>
<div class="line"><a name="l00510"></a><span class="lineno">  510</span>&#160;<span class="comment">//                           workingPool);</span></div>
<div class="line"><a name="l00511"></a><span class="lineno">  511</span>&#160;    }</div>
<div class="line"><a name="l00512"></a><span class="lineno">  512</span>&#160;  }</div>
<div class="ttc" id="classorg_1_1jgap_1_1_genotype_html_a8241b2d89d32b283abe5c9b42974d64d"><div class="ttname"><a href="classorg_1_1jgap_1_1_genotype.html#a8241b2d89d32b283abe5c9b42974d64d">org.jgap.Genotype.getPopulation</a></div><div class="ttdeci">Population getPopulation()</div><div class="ttdef"><b>Definition:</b> Genotype.java:167</div></div>
<div class="ttc" id="classorg_1_1jgap_1_1_genotype_html_a554d1eaf21a0a9437cdbc5b0fbeedef4"><div class="ttname"><a href="classorg_1_1jgap_1_1_genotype.html#a554d1eaf21a0a9437cdbc5b0fbeedef4">org.jgap.Genotype.applyGeneticOperator</a></div><div class="ttdeci">void applyGeneticOperator(GeneticOperator a_operator, Population a_population, List a_chromosomes)</div><div class="ttdef"><b>Definition:</b> Genotype.java:587</div></div>
<div class="ttc" id="classorg_1_1jgap_1_1_genotype_html_ad6795433da5538c6f520253ff9d2382e"><div class="ttname"><a href="classorg_1_1jgap_1_1_genotype.html#ad6795433da5538c6f520253ff9d2382e">org.jgap.Genotype.getChromosomes</a></div><div class="ttdeci">synchronized IChromosome[] getChromosomes()</div><div class="ttdef"><b>Definition:</b> Genotype.java:151</div></div>
</div><!-- fragment -->
</div>
</div>
<a class="anchor" id="a6c14c3a36aa8e4291840f0b9c0facb35"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.Genotype.applyNaturalSelectors </td>
          <td>(</td>
          <td class="paramtype">boolean&#160;</td>
          <td class="paramname"><em>a_processBeforeGeneticOperators</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">protected</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Applies all NaturalSelectors registered with the <a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a>.</p>
<dl class="params"><dt>Parámetros</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_processBeforeGeneticOperators</td><td>true apply NaturalSelectors applicable before GeneticOperators, false: apply the ones applicable after GeneticOperators</td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Autor</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Desde</dt><dd>2.0 </dd></dl>
<dl class="todo"><dt><b><a class="el" href="todo.html#_todo000118">Tareas pendientes:</a></b></dt><dd>optionally use working pool </dd></dl>
<div class="fragment"><div class="line"><a name="l00443"></a><span class="lineno">  443</span>&#160;                                               {</div>
<div class="line"><a name="l00445"></a><span class="lineno">  445</span>&#160;    <span class="keywordflow">try</span> {</div>
<div class="line"><a name="l00446"></a><span class="lineno">  446</span>&#160;      <span class="comment">// Process all natural selectors applicable before executing the</span></div>
<div class="line"><a name="l00447"></a><span class="lineno">  447</span>&#160;      <span class="comment">// genetic operators (reproduction, crossing over, mutation...).</span></div>
<div class="line"><a name="l00448"></a><span class="lineno">  448</span>&#160;      <span class="comment">// -------------------------------------------------------------</span></div>
<div class="line"><a name="l00449"></a><span class="lineno">  449</span>&#160;      <span class="keywordtype">int</span> selectorSize = m_activeConfiguration.getNaturalSelectorsSize(</div>
<div class="line"><a name="l00450"></a><span class="lineno">  450</span>&#160;          a_processBeforeGeneticOperators);</div>
<div class="line"><a name="l00451"></a><span class="lineno">  451</span>&#160;      <span class="keywordflow">if</span> (selectorSize &gt; 0) {</div>
<div class="line"><a name="l00452"></a><span class="lineno">  452</span>&#160;        <span class="keywordtype">int</span> population_size = m_activeConfiguration.getPopulationSize();</div>
<div class="line"><a name="l00453"></a><span class="lineno">  453</span>&#160;        <span class="keywordflow">if</span> (a_processBeforeGeneticOperators) {</div>
<div class="line"><a name="l00454"></a><span class="lineno">  454</span>&#160;          <span class="comment">// Only select part of the previous population into this generation.</span></div>
<div class="line"><a name="l00455"></a><span class="lineno">  455</span>&#160;          <span class="comment">// -----------------------------------------------------------------</span></div>
<div class="line"><a name="l00456"></a><span class="lineno">  456</span>&#160;          population_size = (int) Math.round(population_size *</div>
<div class="line"><a name="l00457"></a><span class="lineno">  457</span>&#160;              <a class="code" href="classorg_1_1jgap_1_1_genotype.html#a786848d2da4fd3d87f9f4caa8f9dd58b">getConfiguration</a>().getSelectFromPrevGen());</div>
<div class="line"><a name="l00458"></a><span class="lineno">  458</span>&#160;        }</div>
<div class="line"><a name="l00459"></a><span class="lineno">  459</span>&#160;        <span class="keywordtype">int</span> single_selection_size;</div>
<div class="line"><a name="l00460"></a><span class="lineno">  460</span>&#160;        Population new_population;</div>
<div class="line"><a name="l00461"></a><span class="lineno">  461</span>&#160;        new_population = <span class="keyword">new</span> Population(m_activeConfiguration,</div>
<div class="line"><a name="l00462"></a><span class="lineno">  462</span>&#160;                                        population_size);</div>
<div class="line"><a name="l00463"></a><span class="lineno">  463</span>&#160;        NaturalSelector selector;</div>
<div class="line"><a name="l00464"></a><span class="lineno">  464</span>&#160;        <span class="comment">// Repopulate the population of chromosomes with those selected</span></div>
<div class="line"><a name="l00465"></a><span class="lineno">  465</span>&#160;        <span class="comment">// by the natural selector. Iterate over all natural selectors.</span></div>
<div class="line"><a name="l00466"></a><span class="lineno">  466</span>&#160;        <span class="comment">// ------------------------------------------------------------</span></div>
<div class="line"><a name="l00467"></a><span class="lineno">  467</span>&#160;        <span class="keywordflow">for</span> (<span class="keywordtype">int</span> i = 0; i &lt; selectorSize; i++) {</div>
<div class="line"><a name="l00468"></a><span class="lineno">  468</span>&#160;          selector = m_activeConfiguration.getNaturalSelector(</div>
<div class="line"><a name="l00469"></a><span class="lineno">  469</span>&#160;              a_processBeforeGeneticOperators, i);</div>
<div class="line"><a name="l00470"></a><span class="lineno">  470</span>&#160;          <span class="keywordflow">if</span> (i == selectorSize - 1 &amp;&amp; i &gt; 0) {</div>
<div class="line"><a name="l00471"></a><span class="lineno">  471</span>&#160;            <span class="comment">// Ensure the last NaturalSelector adds the remaining Chromosomes.</span></div>
<div class="line"><a name="l00472"></a><span class="lineno">  472</span>&#160;            <span class="comment">// ---------------------------------------------------------------</span></div>
<div class="line"><a name="l00473"></a><span class="lineno">  473</span>&#160;            single_selection_size = population_size - <a class="code" href="classorg_1_1jgap_1_1_genotype.html#a8241b2d89d32b283abe5c9b42974d64d">getPopulation</a>().size();</div>
<div class="line"><a name="l00474"></a><span class="lineno">  474</span>&#160;          }</div>
<div class="line"><a name="l00475"></a><span class="lineno">  475</span>&#160;          <span class="keywordflow">else</span> {</div>
<div class="line"><a name="l00476"></a><span class="lineno">  476</span>&#160;            single_selection_size = population_size / selectorSize;</div>
<div class="line"><a name="l00477"></a><span class="lineno">  477</span>&#160;          }</div>
<div class="line"><a name="l00478"></a><span class="lineno">  478</span>&#160;          <span class="comment">// Do selection of Chromosomes.</span></div>
<div class="line"><a name="l00479"></a><span class="lineno">  479</span>&#160;          <span class="comment">// ----------------------------</span></div>
<div class="line"><a name="l00480"></a><span class="lineno">  480</span>&#160;          selector.select(single_selection_size, <a class="code" href="classorg_1_1jgap_1_1_genotype.html#a8241b2d89d32b283abe5c9b42974d64d">getPopulation</a>(),</div>
<div class="line"><a name="l00481"></a><span class="lineno">  481</span>&#160;                          new_population);</div>
<div class="line"><a name="l00482"></a><span class="lineno">  482</span>&#160;          <span class="comment">// Clean up the natural selector.</span></div>
<div class="line"><a name="l00483"></a><span class="lineno">  483</span>&#160;          <span class="comment">// ------------------------------</span></div>
<div class="line"><a name="l00484"></a><span class="lineno">  484</span>&#160;          selector.empty();</div>
<div class="line"><a name="l00485"></a><span class="lineno">  485</span>&#160;        }</div>
<div class="line"><a name="l00486"></a><span class="lineno">  486</span>&#160;        <a class="code" href="classorg_1_1jgap_1_1_genotype.html#ac56a4d112334db07ba333602c4a24706">setPopulation</a>(<span class="keyword">new</span> Population(m_activeConfiguration));</div>
<div class="line"><a name="l00487"></a><span class="lineno">  487</span>&#160;        <a class="code" href="classorg_1_1jgap_1_1_genotype.html#a8241b2d89d32b283abe5c9b42974d64d">getPopulation</a>().addChromosomes(new_population);</div>
<div class="line"><a name="l00488"></a><span class="lineno">  488</span>&#160;      }</div>
<div class="line"><a name="l00489"></a><span class="lineno">  489</span>&#160;    } <span class="keywordflow">catch</span> (InvalidConfigurationException iex) {</div>
<div class="line"><a name="l00490"></a><span class="lineno">  490</span>&#160;      <span class="comment">// This exception should never be reached</span></div>
<div class="line"><a name="l00491"></a><span class="lineno">  491</span>&#160;      <span class="keywordflow">throw</span> <span class="keyword">new</span> IllegalStateException(iex.getMessage());</div>
<div class="line"><a name="l00492"></a><span class="lineno">  492</span>&#160;    }</div>
<div class="line"><a name="l00493"></a><span class="lineno">  493</span>&#160;  }</div>
<div class="ttc" id="classorg_1_1jgap_1_1_genotype_html_a8241b2d89d32b283abe5c9b42974d64d"><div class="ttname"><a href="classorg_1_1jgap_1_1_genotype.html#a8241b2d89d32b283abe5c9b42974d64d">org.jgap.Genotype.getPopulation</a></div><div class="ttdeci">Population getPopulation()</div><div class="ttdef"><b>Definition:</b> Genotype.java:167</div></div>
<div class="ttc" id="classorg_1_1jgap_1_1_genotype_html_a786848d2da4fd3d87f9f4caa8f9dd58b"><div class="ttname"><a href="classorg_1_1jgap_1_1_genotype.html#a786848d2da4fd3d87f9f4caa8f9dd58b">org.jgap.Genotype.getConfiguration</a></div><div class="ttdeci">Configuration getConfiguration()</div><div class="ttdef"><b>Definition:</b> Genotype.java:535</div></div>
<div class="ttc" id="classorg_1_1jgap_1_1_genotype_html_ac56a4d112334db07ba333602c4a24706"><div class="ttname"><a href="classorg_1_1jgap_1_1_genotype.html#ac56a4d112334db07ba333602c4a24706">org.jgap.Genotype.setPopulation</a></div><div class="ttdeci">void setPopulation(Population a_pop)</div><div class="ttdef"><b>Definition:</b> Genotype.java:570</div></div>
</div><!-- fragment -->
</div>
</div>
<a class="anchor" id="ab9bba698920b358bb53131338c9d130a"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">boolean org.jgap.Genotype.equals </td>
          <td>(</td>
          <td class="paramtype">Object&#160;</td>
          <td class="paramname"><em>a_other</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
</div><div class="memdoc">
<p>Compares this <a class="el" href="classorg_1_1jgap_1_1_genotype.html">Genotype</a> against the specified object. The result is true if the argument is an instance of the <a class="el" href="classorg_1_1jgap_1_1_genotype.html">Genotype</a> class, has exactly the same number of chromosomes as the given <a class="el" href="classorg_1_1jgap_1_1_genotype.html">Genotype</a>, and, for each <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> in this <a class="el" href="classorg_1_1jgap_1_1_genotype.html">Genotype</a>, there is an equal chromosome in the given <a class="el" href="classorg_1_1jgap_1_1_genotype.html">Genotype</a>. The chromosomes do not need to appear in the same order within the population.</p>
<dl class="params"><dt>Parámetros</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_other</td><td>the object to compare against </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Devuelve</dt><dd>true if the objects are the same, false otherwise</dd></dl>
<dl class="section author"><dt>Autor</dt><dd>Neil Rotstan </dd>
<dd>
Klaus Meffert </dd></dl>
<dl class="section since"><dt>Desde</dt><dd>1.0 </dd></dl>
<div class="fragment"><div class="line"><a name="l00397"></a><span class="lineno">  397</span>&#160;                                        {</div>
<div class="line"><a name="l00398"></a><span class="lineno">  398</span>&#160;    <span class="keywordflow">try</span> {</div>
<div class="line"><a name="l00399"></a><span class="lineno">  399</span>&#160;      <span class="comment">// First, if the other Genotype is null, then they&#39;re not equal.</span></div>
<div class="line"><a name="l00400"></a><span class="lineno">  400</span>&#160;      <span class="comment">// -------------------------------------------------------------</span></div>
<div class="line"><a name="l00401"></a><span class="lineno">  401</span>&#160;      <span class="keywordflow">if</span> (a_other == null) {</div>
<div class="line"><a name="l00402"></a><span class="lineno">  402</span>&#160;        <span class="keywordflow">return</span> <span class="keyword">false</span>;</div>
<div class="line"><a name="l00403"></a><span class="lineno">  403</span>&#160;      }</div>
<div class="line"><a name="l00404"></a><span class="lineno">  404</span>&#160;      <a class="code" href="classorg_1_1jgap_1_1_genotype.html#a00b11af9a68e9d84bbc52afbf661607f">Genotype</a> otherGenotype = (<a class="code" href="classorg_1_1jgap_1_1_genotype.html#a00b11af9a68e9d84bbc52afbf661607f">Genotype</a>) a_other;</div>
<div class="line"><a name="l00405"></a><span class="lineno">  405</span>&#160;      <span class="comment">// First, make sure the other Genotype has the same number of</span></div>
<div class="line"><a name="l00406"></a><span class="lineno">  406</span>&#160;      <span class="comment">// chromosomes as this one.</span></div>
<div class="line"><a name="l00407"></a><span class="lineno">  407</span>&#160;      <span class="comment">// ----------------------------------------------------------</span></div>
<div class="line"><a name="l00408"></a><span class="lineno">  408</span>&#160;      <span class="keywordflow">if</span> (<a class="code" href="classorg_1_1jgap_1_1_genotype.html#a8241b2d89d32b283abe5c9b42974d64d">getPopulation</a>().size() != otherGenotype.getPopulation().size()) {</div>
<div class="line"><a name="l00409"></a><span class="lineno">  409</span>&#160;        <span class="keywordflow">return</span> <span class="keyword">false</span>;</div>
<div class="line"><a name="l00410"></a><span class="lineno">  410</span>&#160;      }</div>
<div class="line"><a name="l00411"></a><span class="lineno">  411</span>&#160;      <span class="comment">// Next, prepare to compare the chromosomes of the other Genotype</span></div>
<div class="line"><a name="l00412"></a><span class="lineno">  412</span>&#160;      <span class="comment">// against the chromosomes of this Genotype. To make this a lot</span></div>
<div class="line"><a name="l00413"></a><span class="lineno">  413</span>&#160;      <span class="comment">// simpler, we first sort the chromosomes in both this Genotype</span></div>
<div class="line"><a name="l00414"></a><span class="lineno">  414</span>&#160;      <span class="comment">// and the one we&#39;re comparing against. This won&#39;t affect the</span></div>
<div class="line"><a name="l00415"></a><span class="lineno">  415</span>&#160;      <span class="comment">// genetic algorithm (it doesn&#39;t care about the order), but makes</span></div>
<div class="line"><a name="l00416"></a><span class="lineno">  416</span>&#160;      <span class="comment">// it much easier to perform the comparison here.</span></div>
<div class="line"><a name="l00417"></a><span class="lineno">  417</span>&#160;      <span class="comment">// --------------------------------------------------------------</span></div>
<div class="line"><a name="l00418"></a><span class="lineno">  418</span>&#160;      Collections.sort(<a class="code" href="classorg_1_1jgap_1_1_genotype.html#a8241b2d89d32b283abe5c9b42974d64d">getPopulation</a>().getChromosomes());</div>
<div class="line"><a name="l00419"></a><span class="lineno">  419</span>&#160;      Collections.sort(otherGenotype.getPopulation().<a class="code" href="classorg_1_1jgap_1_1_genotype.html#ad6795433da5538c6f520253ff9d2382e">getChromosomes</a>());</div>
<div class="line"><a name="l00420"></a><span class="lineno">  420</span>&#160;      <span class="keywordflow">for</span> (<span class="keywordtype">int</span> i = 0; i &lt; <a class="code" href="classorg_1_1jgap_1_1_genotype.html#a8241b2d89d32b283abe5c9b42974d64d">getPopulation</a>().size(); i++) {</div>
<div class="line"><a name="l00421"></a><span class="lineno">  421</span>&#160;        <span class="keywordflow">if</span> (! (<a class="code" href="classorg_1_1jgap_1_1_genotype.html#a8241b2d89d32b283abe5c9b42974d64d">getPopulation</a>().getChromosome(i).<a class="code" href="classorg_1_1jgap_1_1_genotype.html#ab9bba698920b358bb53131338c9d130a">equals</a>(</div>
<div class="line"><a name="l00422"></a><span class="lineno">  422</span>&#160;            otherGenotype.getPopulation().getChromosome(i)))) {</div>
<div class="line"><a name="l00423"></a><span class="lineno">  423</span>&#160;          <span class="keywordflow">return</span> <span class="keyword">false</span>;</div>
<div class="line"><a name="l00424"></a><span class="lineno">  424</span>&#160;        }</div>
<div class="line"><a name="l00425"></a><span class="lineno">  425</span>&#160;      }</div>
<div class="line"><a name="l00426"></a><span class="lineno">  426</span>&#160;      <span class="keywordflow">return</span> <span class="keyword">true</span>;</div>
<div class="line"><a name="l00427"></a><span class="lineno">  427</span>&#160;    } <span class="keywordflow">catch</span> (ClassCastException e) {</div>
<div class="line"><a name="l00428"></a><span class="lineno">  428</span>&#160;      <span class="keywordflow">return</span> <span class="keyword">false</span>;</div>
<div class="line"><a name="l00429"></a><span class="lineno">  429</span>&#160;    }</div>
<div class="line"><a name="l00430"></a><span class="lineno">  430</span>&#160;  }</div>
<div class="ttc" id="classorg_1_1jgap_1_1_genotype_html_a8241b2d89d32b283abe5c9b42974d64d"><div class="ttname"><a href="classorg_1_1jgap_1_1_genotype.html#a8241b2d89d32b283abe5c9b42974d64d">org.jgap.Genotype.getPopulation</a></div><div class="ttdeci">Population getPopulation()</div><div class="ttdef"><b>Definition:</b> Genotype.java:167</div></div>
<div class="ttc" id="classorg_1_1jgap_1_1_genotype_html_ab9bba698920b358bb53131338c9d130a"><div class="ttname"><a href="classorg_1_1jgap_1_1_genotype.html#ab9bba698920b358bb53131338c9d130a">org.jgap.Genotype.equals</a></div><div class="ttdeci">boolean equals(Object a_other)</div><div class="ttdef"><b>Definition:</b> Genotype.java:397</div></div>
<div class="ttc" id="classorg_1_1jgap_1_1_genotype_html_a00b11af9a68e9d84bbc52afbf661607f"><div class="ttname"><a href="classorg_1_1jgap_1_1_genotype.html#a00b11af9a68e9d84bbc52afbf661607f">org.jgap.Genotype.Genotype</a></div><div class="ttdeci">Genotype(Configuration a_configuration, IChromosome[] a_initialChromosomes)</div><div class="ttdef"><b>Definition:</b> Genotype.java:76</div></div>
<div class="ttc" id="classorg_1_1jgap_1_1_genotype_html_ad6795433da5538c6f520253ff9d2382e"><div class="ttname"><a href="classorg_1_1jgap_1_1_genotype.html#ad6795433da5538c6f520253ff9d2382e">org.jgap.Genotype.getChromosomes</a></div><div class="ttdeci">synchronized IChromosome[] getChromosomes()</div><div class="ttdef"><b>Definition:</b> Genotype.java:151</div></div>
</div><!-- fragment -->
</div>
</div>
<a class="anchor" id="a54a6791fbb6e13920e79f4e2dd719eb4"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">synchronized void org.jgap.Genotype.evolve </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
</div><div class="memdoc">
<p>Evolves the population of Chromosomes within this <a class="el" href="classorg_1_1jgap_1_1_genotype.html">Genotype</a>. This will execute all of the genetic operators added to the present active configuration and then invoke the natural selector to choose which chromosomes will be included in the next generation population. Note that the population size not always remains constant (dependent on the NaturalSelectors used!). To consecutively call this method, use <a class="el" href="classorg_1_1jgap_1_1_genotype.html#ab15d88cf1a6084f9efea2dc935968471">evolve(int)</a>!!!</p>
<dl class="section author"><dt>Autor</dt><dd>Neil Rotstan </dd>
<dd>
Klaus Meffert </dd></dl>
<dl class="section since"><dt>Desde</dt><dd>1.0 </dd></dl>
<div class="fragment"><div class="line"><a name="l00231"></a><span class="lineno">  231</span>&#160;                                    {</div>
<div class="line"><a name="l00232"></a><span class="lineno">  232</span>&#160;    IBreeder breeder = <a class="code" href="classorg_1_1jgap_1_1_genotype.html#a786848d2da4fd3d87f9f4caa8f9dd58b">getConfiguration</a>().getBreeder();</div>
<div class="line"><a name="l00233"></a><span class="lineno">  233</span>&#160;    Population newPop = breeder.evolve(<a class="code" href="classorg_1_1jgap_1_1_genotype.html#a8241b2d89d32b283abe5c9b42974d64d">getPopulation</a>(), <a class="code" href="classorg_1_1jgap_1_1_genotype.html#a786848d2da4fd3d87f9f4caa8f9dd58b">getConfiguration</a>());</div>
<div class="line"><a name="l00234"></a><span class="lineno">  234</span>&#160;    <a class="code" href="classorg_1_1jgap_1_1_genotype.html#ac56a4d112334db07ba333602c4a24706">setPopulation</a>(newPop);</div>
<div class="line"><a name="l00235"></a><span class="lineno">  235</span>&#160;  }</div>
<div class="ttc" id="classorg_1_1jgap_1_1_genotype_html_a8241b2d89d32b283abe5c9b42974d64d"><div class="ttname"><a href="classorg_1_1jgap_1_1_genotype.html#a8241b2d89d32b283abe5c9b42974d64d">org.jgap.Genotype.getPopulation</a></div><div class="ttdeci">Population getPopulation()</div><div class="ttdef"><b>Definition:</b> Genotype.java:167</div></div>
<div class="ttc" id="classorg_1_1jgap_1_1_genotype_html_a786848d2da4fd3d87f9f4caa8f9dd58b"><div class="ttname"><a href="classorg_1_1jgap_1_1_genotype.html#a786848d2da4fd3d87f9f4caa8f9dd58b">org.jgap.Genotype.getConfiguration</a></div><div class="ttdeci">Configuration getConfiguration()</div><div class="ttdef"><b>Definition:</b> Genotype.java:535</div></div>
<div class="ttc" id="classorg_1_1jgap_1_1_genotype_html_ac56a4d112334db07ba333602c4a24706"><div class="ttname"><a href="classorg_1_1jgap_1_1_genotype.html#ac56a4d112334db07ba333602c4a24706">org.jgap.Genotype.setPopulation</a></div><div class="ttdeci">void setPopulation(Population a_pop)</div><div class="ttdef"><b>Definition:</b> Genotype.java:570</div></div>
</div><!-- fragment -->
</div>
</div>
<a class="anchor" id="ab15d88cf1a6084f9efea2dc935968471"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.Genotype.evolve </td>
          <td>(</td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>a_numberOfEvolutions</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
</div><div class="memdoc">
<p>Evolves this <a class="el" href="classorg_1_1jgap_1_1_genotype.html">Genotype</a> the specified number of times. This is equivalent to invoking the standard <a class="el" href="classorg_1_1jgap_1_1_genotype.html#a54a6791fbb6e13920e79f4e2dd719eb4">evolve()</a> method the given number of times in a row.</p>
<dl class="params"><dt>Parámetros</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_numberOfEvolutions</td><td>the number of times to evolve this <a class="el" href="classorg_1_1jgap_1_1_genotype.html">Genotype</a> before returning</td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Autor</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Desde</dt><dd>1.1 </dd></dl>
<div class="fragment"><div class="line"><a name="l00248"></a><span class="lineno">  248</span>&#160;                                               {</div>
<div class="line"><a name="l00249"></a><span class="lineno">  249</span>&#160;    <span class="keywordflow">for</span> (<span class="keywordtype">int</span> i = 0; i &lt; a_numberOfEvolutions; i++) {</div>
<div class="line"><a name="l00250"></a><span class="lineno">  250</span>&#160;      <a class="code" href="classorg_1_1jgap_1_1_genotype.html#a54a6791fbb6e13920e79f4e2dd719eb4">evolve</a>();</div>
<div class="line"><a name="l00251"></a><span class="lineno">  251</span>&#160;    }</div>
<div class="line"><a name="l00252"></a><span class="lineno">  252</span>&#160;    <span class="keywordflow">if</span> (m_activeConfiguration.<a class="code" href="classorg_1_1jgap_1_1_configuration.html#ad41cca8a908a60f0e4575f02f59b780e">isKeepPopulationSizeConstant</a>()) {</div>
<div class="line"><a name="l00253"></a><span class="lineno">  253</span>&#160;      <a class="code" href="classorg_1_1jgap_1_1_genotype.html#af5e224c5b7c05dbe02b3324d5ec16f2a">keepPopSizeConstant</a>(<a class="code" href="classorg_1_1jgap_1_1_genotype.html#a8241b2d89d32b283abe5c9b42974d64d">getPopulation</a>(),</div>
<div class="line"><a name="l00254"></a><span class="lineno">  254</span>&#160;                          m_activeConfiguration.<a class="code" href="classorg_1_1jgap_1_1_configuration.html#a21cd64586b838743538195b6e8a42a11">getPopulationSize</a>());</div>
<div class="line"><a name="l00255"></a><span class="lineno">  255</span>&#160;    }</div>
<div class="line"><a name="l00256"></a><span class="lineno">  256</span>&#160;  }</div>
<div class="ttc" id="classorg_1_1jgap_1_1_genotype_html_a8241b2d89d32b283abe5c9b42974d64d"><div class="ttname"><a href="classorg_1_1jgap_1_1_genotype.html#a8241b2d89d32b283abe5c9b42974d64d">org.jgap.Genotype.getPopulation</a></div><div class="ttdeci">Population getPopulation()</div><div class="ttdef"><b>Definition:</b> Genotype.java:167</div></div>
<div class="ttc" id="classorg_1_1jgap_1_1_configuration_html_ad41cca8a908a60f0e4575f02f59b780e"><div class="ttname"><a href="classorg_1_1jgap_1_1_configuration.html#ad41cca8a908a60f0e4575f02f59b780e">org.jgap.Configuration.isKeepPopulationSizeConstant</a></div><div class="ttdeci">boolean isKeepPopulationSizeConstant()</div><div class="ttdef"><b>Definition:</b> Configuration.java:1462</div></div>
<div class="ttc" id="classorg_1_1jgap_1_1_genotype_html_af5e224c5b7c05dbe02b3324d5ec16f2a"><div class="ttname"><a href="classorg_1_1jgap_1_1_genotype.html#af5e224c5b7c05dbe02b3324d5ec16f2a">org.jgap.Genotype.keepPopSizeConstant</a></div><div class="ttdeci">void keepPopSizeConstant(Population a_pop, int a_maxSize)</div><div class="ttdef"><b>Definition:</b> Genotype.java:602</div></div>
<div class="ttc" id="classorg_1_1jgap_1_1_genotype_html_a54a6791fbb6e13920e79f4e2dd719eb4"><div class="ttname"><a href="classorg_1_1jgap_1_1_genotype.html#a54a6791fbb6e13920e79f4e2dd719eb4">org.jgap.Genotype.evolve</a></div><div class="ttdeci">synchronized void evolve()</div><div class="ttdef"><b>Definition:</b> Genotype.java:231</div></div>
<div class="ttc" id="classorg_1_1jgap_1_1_configuration_html_a21cd64586b838743538195b6e8a42a11"><div class="ttname"><a href="classorg_1_1jgap_1_1_configuration.html#a21cd64586b838743538195b6e8a42a11">org.jgap.Configuration.getPopulationSize</a></div><div class="ttdeci">synchronized int getPopulationSize()</div><div class="ttdef"><b>Definition:</b> Configuration.java:963</div></div>
</div><!-- fragment -->
</div>
</div>
<a class="anchor" id="aa740dd753d31620166979199a38d5a50"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">List&lt;String&gt; org.jgap.Genotype.evolve </td>
          <td>(</td>
          <td class="paramtype">IEvolutionMonitor&#160;</td>
          <td class="paramname"><em>a_monitor</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
</div><div class="memdoc">
<p>Evolves this genotype until the given monitor asks to quit the evolution cycle.</p>
<dl class="params"><dt>Parámetros</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_monitor</td><td>the monitor used to decide when to stop evolution</td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Devuelve</dt><dd>messages of the registered evolution monitor. May indicate why the evolution was asked to be stopped. May be empty, depending on the implementation of the used monitor</dd></dl>
<dl class="section author"><dt>Autor</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Desde</dt><dd>3.4.4 </dd></dl>
<div class="fragment"><div class="line"><a name="l00271"></a><span class="lineno">  271</span>&#160;                                                          {</div>
<div class="line"><a name="l00272"></a><span class="lineno">  272</span>&#160;    a_monitor.start(<a class="code" href="classorg_1_1jgap_1_1_genotype.html#a786848d2da4fd3d87f9f4caa8f9dd58b">getConfiguration</a>());</div>
<div class="line"><a name="l00273"></a><span class="lineno">  273</span>&#160;    <a class="code" href="classorg_1_1jgap_1_1_genotype.html#a786848d2da4fd3d87f9f4caa8f9dd58b">getConfiguration</a>().setMonitor(a_monitor);</div>
<div class="line"><a name="l00274"></a><span class="lineno">  274</span>&#160;    List&lt;String&gt; messages = <span class="keyword">new</span> Vector();</div>
<div class="line"><a name="l00275"></a><span class="lineno">  275</span>&#160;    <span class="keywordflow">do</span> {</div>
<div class="line"><a name="l00276"></a><span class="lineno">  276</span>&#160;      <a class="code" href="classorg_1_1jgap_1_1_genotype.html#a54a6791fbb6e13920e79f4e2dd719eb4">evolve</a>();</div>
<div class="line"><a name="l00277"></a><span class="lineno">  277</span>&#160;      <span class="keywordtype">boolean</span> goon = a_monitor.nextCycle(<a class="code" href="classorg_1_1jgap_1_1_genotype.html#a8241b2d89d32b283abe5c9b42974d64d">getPopulation</a>(), messages);</div>
<div class="line"><a name="l00278"></a><span class="lineno">  278</span>&#160;      <span class="keywordflow">if</span> (!goon) {</div>
<div class="line"><a name="l00279"></a><span class="lineno">  279</span>&#160;        <span class="keywordflow">break</span>;</div>
<div class="line"><a name="l00280"></a><span class="lineno">  280</span>&#160;      }</div>
<div class="line"><a name="l00281"></a><span class="lineno">  281</span>&#160;    } <span class="keywordflow">while</span> (<span class="keyword">true</span>);</div>
<div class="line"><a name="l00282"></a><span class="lineno">  282</span>&#160;    <span class="keywordflow">return</span> messages;</div>
<div class="line"><a name="l00283"></a><span class="lineno">  283</span>&#160;  }</div>
<div class="ttc" id="classorg_1_1jgap_1_1_genotype_html_a8241b2d89d32b283abe5c9b42974d64d"><div class="ttname"><a href="classorg_1_1jgap_1_1_genotype.html#a8241b2d89d32b283abe5c9b42974d64d">org.jgap.Genotype.getPopulation</a></div><div class="ttdeci">Population getPopulation()</div><div class="ttdef"><b>Definition:</b> Genotype.java:167</div></div>
<div class="ttc" id="classorg_1_1jgap_1_1_genotype_html_a786848d2da4fd3d87f9f4caa8f9dd58b"><div class="ttname"><a href="classorg_1_1jgap_1_1_genotype.html#a786848d2da4fd3d87f9f4caa8f9dd58b">org.jgap.Genotype.getConfiguration</a></div><div class="ttdeci">Configuration getConfiguration()</div><div class="ttdef"><b>Definition:</b> Genotype.java:535</div></div>
<div class="ttc" id="classorg_1_1jgap_1_1_genotype_html_a54a6791fbb6e13920e79f4e2dd719eb4"><div class="ttname"><a href="classorg_1_1jgap_1_1_genotype.html#a54a6791fbb6e13920e79f4e2dd719eb4">org.jgap.Genotype.evolve</a></div><div class="ttdeci">synchronized void evolve()</div><div class="ttdef"><b>Definition:</b> Genotype.java:231</div></div>
</div><!-- fragment -->
</div>
</div>
<a class="anchor" id="ac622bbf487c0ab7344d46ca669607cef"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.Genotype.fillPopulation </td>
          <td>(</td>
          <td class="paramtype">final int&#160;</td>
          <td class="paramname"><em>a_num</em></td><td>)</td>
          <td> throws <a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a></td>
        </tr>
      </table>
</div><div class="memdoc">
<p>Fills up the population with random chromosomes if necessary.</p>
<dl class="params"><dt>Parámetros</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_num</td><td>the number of chromosomes to add </td></tr>
  </table>
  </dd>
</dl>
<dl class="exception"><dt>Excepciones</dt><dd>
  <table class="exception">
    <tr><td class="paramname"><a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a></td><td></td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Autor</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Desde</dt><dd>3.2 </dd></dl>
<div class="fragment"><div class="line"><a name="l00355"></a><span class="lineno">  355</span>&#160;                                           {</div>
<div class="line"><a name="l00356"></a><span class="lineno">  356</span>&#160;    IChromosome sampleChrom = <a class="code" href="classorg_1_1jgap_1_1_genotype.html#a786848d2da4fd3d87f9f4caa8f9dd58b">getConfiguration</a>().getSampleChromosome();</div>
<div class="line"><a name="l00357"></a><span class="lineno">  357</span>&#160;    Class sampleClass = sampleChrom.getClass();</div>
<div class="line"><a name="l00358"></a><span class="lineno">  358</span>&#160;    IInitializer chromIniter = <a class="code" href="classorg_1_1jgap_1_1_genotype.html#a786848d2da4fd3d87f9f4caa8f9dd58b">getConfiguration</a>().getJGAPFactory().</div>
<div class="line"><a name="l00359"></a><span class="lineno">  359</span>&#160;        getInitializerFor(sampleChrom, sampleClass);</div>
<div class="line"><a name="l00360"></a><span class="lineno">  360</span>&#160;    <span class="keywordflow">if</span> (chromIniter == null) {</div>
<div class="line"><a name="l00361"></a><span class="lineno">  361</span>&#160;      <span class="keywordflow">throw</span> <span class="keyword">new</span> InvalidConfigurationException(<span class="stringliteral">&quot;No initializer found for class &quot;</span></div>
<div class="line"><a name="l00362"></a><span class="lineno">  362</span>&#160;          + sampleClass);</div>
<div class="line"><a name="l00363"></a><span class="lineno">  363</span>&#160;    }</div>
<div class="line"><a name="l00364"></a><span class="lineno">  364</span>&#160;    <span class="keywordflow">try</span> {</div>
<div class="line"><a name="l00365"></a><span class="lineno">  365</span>&#160;      <span class="keywordflow">for</span> (<span class="keywordtype">int</span> i = 0; i &lt; a_num; i++) {</div>
<div class="line"><a name="l00366"></a><span class="lineno">  366</span>&#160;        <a class="code" href="classorg_1_1jgap_1_1_genotype.html#a8241b2d89d32b283abe5c9b42974d64d">getPopulation</a>().addChromosome( (IChromosome) chromIniter.perform(</div>
<div class="line"><a name="l00367"></a><span class="lineno">  367</span>&#160;            sampleChrom,</div>
<div class="line"><a name="l00368"></a><span class="lineno">  368</span>&#160;            sampleClass, null));</div>
<div class="line"><a name="l00369"></a><span class="lineno">  369</span>&#160;      }</div>
<div class="line"><a name="l00370"></a><span class="lineno">  370</span>&#160;    } <span class="keywordflow">catch</span> (Exception ex) {</div>
<div class="line"><a name="l00371"></a><span class="lineno">  371</span>&#160;      <span class="comment">// Try to propagate exception, see &quot;bug&quot; 1661635.</span></div>
<div class="line"><a name="l00372"></a><span class="lineno">  372</span>&#160;      <span class="comment">// ----------------------------------------------</span></div>
<div class="line"><a name="l00373"></a><span class="lineno">  373</span>&#160;      <span class="keywordflow">if</span> (ex.getCause() != null) {</div>
<div class="line"><a name="l00374"></a><span class="lineno">  374</span>&#160;        <span class="keywordflow">throw</span> <span class="keyword">new</span> IllegalStateException(ex.getCause().toString());</div>
<div class="line"><a name="l00375"></a><span class="lineno">  375</span>&#160;      }</div>
<div class="line"><a name="l00376"></a><span class="lineno">  376</span>&#160;      <span class="keywordflow">else</span> {</div>
<div class="line"><a name="l00377"></a><span class="lineno">  377</span>&#160;        <span class="keywordflow">throw</span> <span class="keyword">new</span> IllegalStateException(ex.getMessage());</div>
<div class="line"><a name="l00378"></a><span class="lineno">  378</span>&#160;      }</div>
<div class="line"><a name="l00379"></a><span class="lineno">  379</span>&#160;    }</div>
<div class="line"><a name="l00380"></a><span class="lineno">  380</span>&#160;  }</div>
<div class="ttc" id="classorg_1_1jgap_1_1_genotype_html_a8241b2d89d32b283abe5c9b42974d64d"><div class="ttname"><a href="classorg_1_1jgap_1_1_genotype.html#a8241b2d89d32b283abe5c9b42974d64d">org.jgap.Genotype.getPopulation</a></div><div class="ttdeci">Population getPopulation()</div><div class="ttdef"><b>Definition:</b> Genotype.java:167</div></div>
<div class="ttc" id="classorg_1_1jgap_1_1_genotype_html_a786848d2da4fd3d87f9f4caa8f9dd58b"><div class="ttname"><a href="classorg_1_1jgap_1_1_genotype.html#a786848d2da4fd3d87f9f4caa8f9dd58b">org.jgap.Genotype.getConfiguration</a></div><div class="ttdeci">Configuration getConfiguration()</div><div class="ttdef"><b>Definition:</b> Genotype.java:535</div></div>
</div><!-- fragment -->
</div>
</div>
<a class="anchor" id="ad6795433da5538c6f520253ff9d2382e"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">synchronized <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html">IChromosome</a> [] org.jgap.Genotype.getChromosomes </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
</div><div class="memdoc">
<p>Retrieves the array of Chromosomes that make up the population of this <a class="el" href="classorg_1_1jgap_1_1_genotype.html">Genotype</a> instance.</p>
<dl class="section return"><dt>Devuelve</dt><dd>the <a class="el" href="classorg_1_1jgap_1_1_population.html">Population</a> of Chromosomes</dd></dl>
<dl class="section author"><dt>Autor</dt><dd>Neil Rotstan </dd>
<dd>
Klaus Meffert </dd></dl>
<dl class="section since"><dt>Desde</dt><dd>1.0 </dd></dl>
<dl class="deprecated"><dt><b><a class="el" href="deprecated.html#_deprecated000008">Obsoleto:</a></b></dt><dd>uses <a class="el" href="classorg_1_1jgap_1_1_genotype.html#a8241b2d89d32b283abe5c9b42974d64d">getPopulation()</a> instead </dd></dl>
<div class="fragment"><div class="line"><a name="l00151"></a><span class="lineno">  151</span>&#160;                                                     {</div>
<div class="line"><a name="l00152"></a><span class="lineno">  152</span>&#160;    Iterator it = <a class="code" href="classorg_1_1jgap_1_1_genotype.html#a8241b2d89d32b283abe5c9b42974d64d">getPopulation</a>().iterator();</div>
<div class="line"><a name="l00153"></a><span class="lineno">  153</span>&#160;    IChromosome[] result = <span class="keyword">new</span> Chromosome[<a class="code" href="classorg_1_1jgap_1_1_genotype.html#a8241b2d89d32b283abe5c9b42974d64d">getPopulation</a>().size()];</div>
<div class="line"><a name="l00154"></a><span class="lineno">  154</span>&#160;    <span class="keywordtype">int</span> i = 0;</div>
<div class="line"><a name="l00155"></a><span class="lineno">  155</span>&#160;    <span class="keywordflow">while</span> (it.hasNext()) {</div>
<div class="line"><a name="l00156"></a><span class="lineno">  156</span>&#160;      result[i++] = (IChromosome) it.next();</div>
<div class="line"><a name="l00157"></a><span class="lineno">  157</span>&#160;    }</div>
<div class="line"><a name="l00158"></a><span class="lineno">  158</span>&#160;    <span class="keywordflow">return</span> result;</div>
<div class="line"><a name="l00159"></a><span class="lineno">  159</span>&#160;  }</div>
<div class="ttc" id="classorg_1_1jgap_1_1_genotype_html_a8241b2d89d32b283abe5c9b42974d64d"><div class="ttname"><a href="classorg_1_1jgap_1_1_genotype.html#a8241b2d89d32b283abe5c9b42974d64d">org.jgap.Genotype.getPopulation</a></div><div class="ttdeci">Population getPopulation()</div><div class="ttdef"><b>Definition:</b> Genotype.java:167</div></div>
</div><!-- fragment -->
</div>
</div>
<a class="anchor" id="a786848d2da4fd3d87f9f4caa8f9dd58b"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a> org.jgap.Genotype.getConfiguration </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
</div><div class="memdoc">
<div class="fragment"><div class="line"><a name="l00535"></a><span class="lineno">  535</span>&#160;                                          {</div>
<div class="line"><a name="l00536"></a><span class="lineno">  536</span>&#160;    <span class="keywordflow">return</span> m_activeConfiguration;</div>
<div class="line"><a name="l00537"></a><span class="lineno">  537</span>&#160;  }</div>
</div><!-- fragment -->
</div>
</div>
<a class="anchor" id="a84fd1b399188ea10c0af4b7c1152e0c2"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">List&lt;IEvolveJob&gt; org.jgap.Genotype.getEvolves </td>
          <td>(</td>
          <td class="paramtype">IPopulationSplitter&#160;</td>
          <td class="paramname"><em>a_splitter</em></td><td>)</td>
          <td> throws Exception</td>
        </tr>
      </table>
</div><div class="memdoc">
<p>Splits a population into pieces that can be evolved independently.</p>
<dl class="params"><dt>Parámetros</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_splitter</td><td>splits the population </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Devuelve</dt><dd>list of IEvolveJob objects </dd></dl>
<dl class="exception"><dt>Excepciones</dt><dd>
  <table class="exception">
    <tr><td class="paramname">Exception</td><td></td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Autor</dt><dd>Klaus Meffert </dd></dl>
<div class="fragment"><div class="line"><a name="l00645"></a><span class="lineno">  645</span>&#160;                       {</div>
<div class="line"><a name="l00646"></a><span class="lineno">  646</span>&#160;    <span class="comment">// We return a list of IEvolveJob instances.</span></div>
<div class="line"><a name="l00647"></a><span class="lineno">  647</span>&#160;    <span class="comment">// -----------------------------------------</span></div>
<div class="line"><a name="l00648"></a><span class="lineno">  648</span>&#160;    List result = <span class="keyword">new</span> Vector();</div>
<div class="line"><a name="l00649"></a><span class="lineno">  649</span>&#160;    Population[] pops = a_splitter.split(<a class="code" href="classorg_1_1jgap_1_1_genotype.html#a8241b2d89d32b283abe5c9b42974d64d">getPopulation</a>());</div>
<div class="line"><a name="l00650"></a><span class="lineno">  650</span>&#160;    <span class="comment">// Feed the population chunks into different evolve jobs.</span></div>
<div class="line"><a name="l00651"></a><span class="lineno">  651</span>&#160;    <span class="comment">// ------------------------------------------------------</span></div>
<div class="line"><a name="l00652"></a><span class="lineno">  652</span>&#160;    <span class="keywordflow">for</span> (<span class="keywordtype">int</span> i = 0; i &lt; pops.length; i++) {</div>
<div class="line"><a name="l00653"></a><span class="lineno">  653</span>&#160;      Configuration newConf = (Configuration) <a class="code" href="classorg_1_1jgap_1_1_genotype.html#a786848d2da4fd3d87f9f4caa8f9dd58b">getConfiguration</a>().clone();</div>
<div class="line"><a name="l00654"></a><span class="lineno">  654</span>&#160;      <span class="comment">// Adjust population size.</span></div>
<div class="line"><a name="l00655"></a><span class="lineno">  655</span>&#160;      <span class="comment">// -----------------------</span></div>
<div class="line"><a name="l00656"></a><span class="lineno">  656</span>&#160;      <span class="comment">/*@todo impl.*/</span></div>
<div class="line"><a name="l00657"></a><span class="lineno">  657</span>&#160;      EvolveData data = <span class="keyword">new</span> EvolveData(newConf);</div>
<div class="line"><a name="l00658"></a><span class="lineno">  658</span>&#160;      <span class="keywordflow">if</span> (pops[i] == null) {</div>
<div class="line"><a name="l00659"></a><span class="lineno">  659</span>&#160;        <span class="keywordflow">throw</span> <span class="keyword">new</span> IllegalStateException(<span class="stringliteral">&quot;Population must no be null&quot;</span></div>
<div class="line"><a name="l00660"></a><span class="lineno">  660</span>&#160;                                        + <span class="stringliteral">&quot; (Index: &quot;</span> + i</div>
<div class="line"><a name="l00661"></a><span class="lineno">  661</span>&#160;                                        + <span class="stringliteral">&quot;, Splitter: &quot;</span></div>
<div class="line"><a name="l00662"></a><span class="lineno">  662</span>&#160;                                        + a_splitter.getClass().getName() + <span class="stringliteral">&quot;)&quot;</span>);</div>
<div class="line"><a name="l00663"></a><span class="lineno">  663</span>&#160;      }</div>
<div class="line"><a name="l00664"></a><span class="lineno">  664</span>&#160;      data.setPopulation(pops[i]);</div>
<div class="line"><a name="l00665"></a><span class="lineno">  665</span>&#160;      data.setBreeder(newConf.getBreeder());</div>
<div class="line"><a name="l00666"></a><span class="lineno">  666</span>&#160;      IEvolveJob evolver = <span class="keyword">new</span> EvolveJob(data);</div>
<div class="line"><a name="l00667"></a><span class="lineno">  667</span>&#160;      result.add(evolver);</div>
<div class="line"><a name="l00668"></a><span class="lineno">  668</span>&#160;    }</div>
<div class="line"><a name="l00669"></a><span class="lineno">  669</span>&#160;    <a class="code" href="classorg_1_1jgap_1_1_genotype.html#a8241b2d89d32b283abe5c9b42974d64d">getPopulation</a>().clear();</div>
<div class="line"><a name="l00670"></a><span class="lineno">  670</span>&#160;    <span class="keywordflow">return</span> result;</div>
<div class="line"><a name="l00671"></a><span class="lineno">  671</span>&#160;  }</div>
<div class="ttc" id="classorg_1_1jgap_1_1_genotype_html_a8241b2d89d32b283abe5c9b42974d64d"><div class="ttname"><a href="classorg_1_1jgap_1_1_genotype.html#a8241b2d89d32b283abe5c9b42974d64d">org.jgap.Genotype.getPopulation</a></div><div class="ttdeci">Population getPopulation()</div><div class="ttdef"><b>Definition:</b> Genotype.java:167</div></div>
<div class="ttc" id="classorg_1_1jgap_1_1_genotype_html_a786848d2da4fd3d87f9f4caa8f9dd58b"><div class="ttname"><a href="classorg_1_1jgap_1_1_genotype.html#a786848d2da4fd3d87f9f4caa8f9dd58b">org.jgap.Genotype.getConfiguration</a></div><div class="ttdeci">Configuration getConfiguration()</div><div class="ttdef"><b>Definition:</b> Genotype.java:535</div></div>
</div><!-- fragment -->
</div>
</div>
<a class="anchor" id="a05380af4c0737841c09d3a5533f9f942"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">synchronized <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html">IChromosome</a> org.jgap.Genotype.getFittestChromosome </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
</div><div class="memdoc">
<p>Retrieves the <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> in the <a class="el" href="classorg_1_1jgap_1_1_population.html">Population</a> with the highest fitness value.</p>
<dl class="section return"><dt>Devuelve</dt><dd>the <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> with the highest fitness value, or null if there are no chromosomes in this <a class="el" href="classorg_1_1jgap_1_1_genotype.html">Genotype</a></dd></dl>
<dl class="section author"><dt>Autor</dt><dd>Neil Rotstan </dd>
<dd>
Klaus Meffert </dd></dl>
<dl class="section since"><dt>Desde</dt><dd>1.0 </dd></dl>
<div class="fragment"><div class="line"><a name="l00182"></a><span class="lineno">  182</span>&#160;                                                         {</div>
<div class="line"><a name="l00183"></a><span class="lineno">  183</span>&#160;    <span class="keywordflow">return</span> <a class="code" href="classorg_1_1jgap_1_1_genotype.html#a8241b2d89d32b283abe5c9b42974d64d">getPopulation</a>().determineFittestChromosome();</div>
<div class="line"><a name="l00184"></a><span class="lineno">  184</span>&#160;  }</div>
<div class="ttc" id="classorg_1_1jgap_1_1_genotype_html_a8241b2d89d32b283abe5c9b42974d64d"><div class="ttname"><a href="classorg_1_1jgap_1_1_genotype.html#a8241b2d89d32b283abe5c9b42974d64d">org.jgap.Genotype.getPopulation</a></div><div class="ttdeci">Population getPopulation()</div><div class="ttdef"><b>Definition:</b> Genotype.java:167</div></div>
</div><!-- fragment -->
</div>
</div>
<a class="anchor" id="a6c40835533352287db8d34e5fae6875f"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">synchronized <a class="el" href="interfaceorg_1_1jgap_1_1_i_chromosome.html">IChromosome</a> org.jgap.Genotype.getFittestChromosome </td>
          <td>(</td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>a_startIndex</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>a_endIndex</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">
<p>Retrieves the <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> in the <a class="el" href="classorg_1_1jgap_1_1_population.html">Population</a> with the highest fitness value within the given indices.</p>
<dl class="params"><dt>Parámetros</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_startIndex</td><td>the index to start the determination with </td></tr>
    <tr><td class="paramname">a_endIndex</td><td>the index to end the determination with </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Devuelve</dt><dd>the <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> with the highest fitness value within the given indices, or null if there are no chromosomes in this <a class="el" href="classorg_1_1jgap_1_1_genotype.html">Genotype</a></dd></dl>
<dl class="section author"><dt>Autor</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Desde</dt><dd>3.0 </dd></dl>
<div class="fragment"><div class="line"><a name="l00199"></a><span class="lineno">  199</span>&#160;                      {</div>
<div class="line"><a name="l00200"></a><span class="lineno">  200</span>&#160;    <span class="keywordflow">return</span> <a class="code" href="classorg_1_1jgap_1_1_genotype.html#a8241b2d89d32b283abe5c9b42974d64d">getPopulation</a>().determineFittestChromosome(a_startIndex, a_endIndex);</div>
<div class="line"><a name="l00201"></a><span class="lineno">  201</span>&#160;  }</div>
<div class="ttc" id="classorg_1_1jgap_1_1_genotype_html_a8241b2d89d32b283abe5c9b42974d64d"><div class="ttname"><a href="classorg_1_1jgap_1_1_genotype.html#a8241b2d89d32b283abe5c9b42974d64d">org.jgap.Genotype.getPopulation</a></div><div class="ttdeci">Population getPopulation()</div><div class="ttdef"><b>Definition:</b> Genotype.java:167</div></div>
</div><!-- fragment -->
</div>
</div>
<a class="anchor" id="a602d6b1413007ef28880cf88ccd2f912"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">synchronized List org.jgap.Genotype.getFittestChromosomes </td>
          <td>(</td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>a_numberOfChromosomes</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
</div><div class="memdoc">
<p>Retrieves the top n Chromsomes in the population (the ones with the best fitness values).</p>
<dl class="params"><dt>Parámetros</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_numberOfChromosomes</td><td>the number of chromosomes desired </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Devuelve</dt><dd>the list of Chromosomes with the highest fitness values, or null if there are no chromosomes in this <a class="el" href="classorg_1_1jgap_1_1_genotype.html">Genotype</a></dd></dl>
<dl class="section author"><dt>Autor</dt><dd>Charles Kevin Hill </dd></dl>
<dl class="section since"><dt>Desde</dt><dd>2.4 </dd></dl>
<div class="fragment"><div class="line"><a name="l00214"></a><span class="lineno">  214</span>&#160;                                                                            {</div>
<div class="line"><a name="l00215"></a><span class="lineno">  215</span>&#160;    <span class="keywordflow">return</span> <a class="code" href="classorg_1_1jgap_1_1_genotype.html#a8241b2d89d32b283abe5c9b42974d64d">getPopulation</a>().determineFittestChromosomes(a_numberOfChromosomes);</div>
<div class="line"><a name="l00216"></a><span class="lineno">  216</span>&#160;  }</div>
<div class="ttc" id="classorg_1_1jgap_1_1_genotype_html_a8241b2d89d32b283abe5c9b42974d64d"><div class="ttname"><a href="classorg_1_1jgap_1_1_genotype.html#a8241b2d89d32b283abe5c9b42974d64d">org.jgap.Genotype.getPopulation</a></div><div class="ttdeci">Population getPopulation()</div><div class="ttdef"><b>Definition:</b> Genotype.java:167</div></div>
</div><!-- fragment -->
</div>
</div>
<a class="anchor" id="a8241b2d89d32b283abe5c9b42974d64d"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="classorg_1_1jgap_1_1_population.html">Population</a> org.jgap.Genotype.getPopulation </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
</div><div class="memdoc">
<dl class="section return"><dt>Devuelve</dt><dd>the current population of chromosomes</dd></dl>
<dl class="section author"><dt>Autor</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Desde</dt><dd>2.1 </dd></dl>
<div class="fragment"><div class="line"><a name="l00167"></a><span class="lineno">  167</span>&#160;                                    {</div>
<div class="line"><a name="l00168"></a><span class="lineno">  168</span>&#160;    <span class="keywordflow">return</span> m_population;</div>
<div class="line"><a name="l00169"></a><span class="lineno">  169</span>&#160;  }</div>
</div><!-- fragment -->
</div>
</div>
<a class="anchor" id="aaf3d4bf0b470aa4655e85262289443eb"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">static <a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a> org.jgap.Genotype.getStaticConfiguration </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">static</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<dl class="section return"><dt>Devuelve</dt><dd>the configuration to use with the Genetic Algorithm</dd></dl>
<dl class="section author"><dt>Autor</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Desde</dt><dd>2.0 </dd></dl>
<div class="fragment"><div class="line"><a name="l00520"></a><span class="lineno">  520</span>&#160;                                                       {</div>
<div class="line"><a name="l00521"></a><span class="lineno">  521</span>&#160;    <span class="keywordflow">return</span> m_staticConfiguration;</div>
<div class="line"><a name="l00522"></a><span class="lineno">  522</span>&#160;  }</div>
</div><!-- fragment -->
</div>
</div>
<a class="anchor" id="abe044a5473e935c2093ee3c6c18775f7"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">int org.jgap.Genotype.hashCode </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
</div><div class="memdoc">
<div class="fragment"><div class="line"><a name="l00555"></a><span class="lineno">  555</span>&#160;                        {</div>
<div class="line"><a name="l00556"></a><span class="lineno">  556</span>&#160;    <span class="keywordtype">int</span> i, size = <a class="code" href="classorg_1_1jgap_1_1_genotype.html#a8241b2d89d32b283abe5c9b42974d64d">getPopulation</a>().size();</div>
<div class="line"><a name="l00557"></a><span class="lineno">  557</span>&#160;    IChromosome s;</div>
<div class="line"><a name="l00558"></a><span class="lineno">  558</span>&#160;    <span class="keywordtype">int</span> twopower = 1;</div>
<div class="line"><a name="l00559"></a><span class="lineno">  559</span>&#160;    <span class="comment">// For empty genotype we want a special value different from other hashcode</span></div>
<div class="line"><a name="l00560"></a><span class="lineno">  560</span>&#160;    <span class="comment">// implementations.</span></div>
<div class="line"><a name="l00561"></a><span class="lineno">  561</span>&#160;    <span class="comment">// ------------------------------------------------------------------------</span></div>
<div class="line"><a name="l00562"></a><span class="lineno">  562</span>&#160;    <span class="keywordtype">int</span> localHashCode = -573;</div>
<div class="line"><a name="l00563"></a><span class="lineno">  563</span>&#160;    <span class="keywordflow">for</span> (i = 0; i &lt; size; i++, twopower = 2 * twopower) {</div>
<div class="line"><a name="l00564"></a><span class="lineno">  564</span>&#160;      s = <a class="code" href="classorg_1_1jgap_1_1_genotype.html#a8241b2d89d32b283abe5c9b42974d64d">getPopulation</a>().getChromosome(i);</div>
<div class="line"><a name="l00565"></a><span class="lineno">  565</span>&#160;      localHashCode = 31 * localHashCode + s.hashCode();</div>
<div class="line"><a name="l00566"></a><span class="lineno">  566</span>&#160;    }</div>
<div class="line"><a name="l00567"></a><span class="lineno">  567</span>&#160;    <span class="keywordflow">return</span> localHashCode;</div>
<div class="line"><a name="l00568"></a><span class="lineno">  568</span>&#160;  }</div>
<div class="ttc" id="classorg_1_1jgap_1_1_genotype_html_a8241b2d89d32b283abe5c9b42974d64d"><div class="ttname"><a href="classorg_1_1jgap_1_1_genotype.html#a8241b2d89d32b283abe5c9b42974d64d">org.jgap.Genotype.getPopulation</a></div><div class="ttdeci">Population getPopulation()</div><div class="ttdef"><b>Definition:</b> Genotype.java:167</div></div>
</div><!-- fragment -->
</div>
</div>
<a class="anchor" id="af5e224c5b7c05dbe02b3324d5ec16f2a"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.Genotype.keepPopSizeConstant </td>
          <td>(</td>
          <td class="paramtype"><a class="el" href="classorg_1_1jgap_1_1_population.html">Population</a>&#160;</td>
          <td class="paramname"><em>a_pop</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">int&#160;</td>
          <td class="paramname"><em>a_maxSize</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">protected</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Cares that the population size does not exceed the given maximum size.</p>
<dl class="params"><dt>Parámetros</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_pop</td><td>the population to keep constant in size </td></tr>
    <tr><td class="paramname">a_maxSize</td><td>the maximum size allowed for the population</td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Autor</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Desde</dt><dd>2.5 </dd></dl>
<dl class="todo"><dt><b><a class="el" href="todo.html#_todo000119">Tareas pendientes:</a></b></dt><dd>use StandardPostSelector instead? </dd></dl>
<div class="fragment"><div class="line"><a name="l00602"></a><span class="lineno">  602</span>&#160;                                                                      {</div>
<div class="line"><a name="l00604"></a><span class="lineno">  604</span>&#160;    <span class="keywordtype">int</span> popSize = a_pop.size();</div>
<div class="line"><a name="l00605"></a><span class="lineno">  605</span>&#160;    <span class="comment">// See request  1213752.</span></div>
<div class="line"><a name="l00606"></a><span class="lineno">  606</span>&#160;    <span class="comment">// ---------------------</span></div>
<div class="line"><a name="l00607"></a><span class="lineno">  607</span>&#160;    <span class="keywordflow">while</span> (popSize &gt; a_maxSize) {</div>
<div class="line"><a name="l00608"></a><span class="lineno">  608</span>&#160;      <span class="comment">// Remove a chromosome.</span></div>
<div class="line"><a name="l00609"></a><span class="lineno">  609</span>&#160;      <span class="comment">// --------------------</span></div>
<div class="line"><a name="l00610"></a><span class="lineno">  610</span>&#160;      a_pop.removeChromosome(0);</div>
<div class="line"><a name="l00611"></a><span class="lineno">  611</span>&#160;      popSize--;</div>
<div class="line"><a name="l00612"></a><span class="lineno">  612</span>&#160;    }</div>
<div class="line"><a name="l00613"></a><span class="lineno">  613</span>&#160;  }</div>
</div><!-- fragment -->
</div>
</div>
<a class="anchor" id="aa2b0341edc420ad61e797508b8bf4dd5"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.Genotype.mergeResults </td>
          <td>(</td>
          <td class="paramtype">IPopulationMerger&#160;</td>
          <td class="paramname"><em>a_merger</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">EvolveResult[]&#160;</td>
          <td class="paramname"><em>a_results</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td> throws Exception</td>
        </tr>
      </table>
</div><div class="memdoc">
<dl class="todo"><dt><b><a class="el" href="todo.html#_todo000120">Tareas pendientes:</a></b></dt><dd>use/enhance IPopulationMerger </dd></dl>
<div class="fragment"><div class="line"><a name="l00674"></a><span class="lineno">  674</span>&#160;                       {</div>
<div class="line"><a name="l00675"></a><span class="lineno">  675</span>&#160;    <span class="keywordtype">int</span> size = a_results.length;</div>
<div class="line"><a name="l00676"></a><span class="lineno">  676</span>&#160;    Population target = <span class="keyword">new</span> Population(<a class="code" href="classorg_1_1jgap_1_1_genotype.html#a786848d2da4fd3d87f9f4caa8f9dd58b">getConfiguration</a>());</div>
<div class="line"><a name="l00677"></a><span class="lineno">  677</span>&#160;    <span class="keywordflow">for</span> (<span class="keywordtype">int</span> i = 0; i &lt; size; i++) {</div>
<div class="line"><a name="l00678"></a><span class="lineno">  678</span>&#160;      EvolveResult singleResult = a_results[i];</div>
<div class="line"><a name="l00679"></a><span class="lineno">  679</span>&#160;      <span class="keywordflow">if</span> (singleResult == null) {</div>
<div class="line"><a name="l00680"></a><span class="lineno">  680</span>&#160;        <span class="keywordflow">throw</span> <span class="keyword">new</span> IllegalStateException(<span class="stringliteral">&quot;Single result is null!&quot;</span>);</div>
<div class="line"><a name="l00681"></a><span class="lineno">  681</span>&#160;      }</div>
<div class="line"><a name="l00682"></a><span class="lineno">  682</span>&#160;      Population pop = singleResult.getPopulation();</div>
<div class="line"><a name="l00684"></a><span class="lineno">  684</span>&#160;<span class="comment">//      a_merger.mergePopulations()</span></div>
<div class="line"><a name="l00685"></a><span class="lineno">  685</span>&#160;      List goodOnes = pop.determineFittestChromosomes(3);</div>
<div class="line"><a name="l00686"></a><span class="lineno">  686</span>&#160;      <span class="keywordflow">for</span> (<span class="keywordtype">int</span> j = 0; j &lt; goodOnes.size(); j++) {</div>
<div class="line"><a name="l00687"></a><span class="lineno">  687</span>&#160;        IChromosome goodOne = (IChromosome) goodOnes.get(j);</div>
<div class="line"><a name="l00688"></a><span class="lineno">  688</span>&#160;        target.addChromosome(goodOne);</div>
<div class="line"><a name="l00689"></a><span class="lineno">  689</span>&#160;      }</div>
<div class="line"><a name="l00690"></a><span class="lineno">  690</span>&#160;    }</div>
<div class="line"><a name="l00691"></a><span class="lineno">  691</span>&#160;    <a class="code" href="classorg_1_1jgap_1_1_genotype.html#ac56a4d112334db07ba333602c4a24706">setPopulation</a>(target);</div>
<div class="line"><a name="l00692"></a><span class="lineno">  692</span>&#160;  }</div>
<div class="ttc" id="classorg_1_1jgap_1_1_genotype_html_a786848d2da4fd3d87f9f4caa8f9dd58b"><div class="ttname"><a href="classorg_1_1jgap_1_1_genotype.html#a786848d2da4fd3d87f9f4caa8f9dd58b">org.jgap.Genotype.getConfiguration</a></div><div class="ttdeci">Configuration getConfiguration()</div><div class="ttdef"><b>Definition:</b> Genotype.java:535</div></div>
<div class="ttc" id="classorg_1_1jgap_1_1_genotype_html_ac56a4d112334db07ba333602c4a24706"><div class="ttname"><a href="classorg_1_1jgap_1_1_genotype.html#ac56a4d112334db07ba333602c4a24706">org.jgap.Genotype.setPopulation</a></div><div class="ttdeci">void setPopulation(Population a_pop)</div><div class="ttdef"><b>Definition:</b> Genotype.java:570</div></div>
</div><!-- fragment -->
</div>
</div>
<a class="anchor" id="a930184e2920b4642f20d0b8bc8412005"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">static <a class="el" href="classorg_1_1jgap_1_1_genotype.html">Genotype</a> org.jgap.Genotype.randomInitialGenotype </td>
          <td>(</td>
          <td class="paramtype"><a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a>&#160;</td>
          <td class="paramname"><em>a_configuration</em></td><td>)</td>
          <td> throws <a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">static</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Convenience method that returns a newly constructed <a class="el" href="classorg_1_1jgap_1_1_genotype.html">Genotype</a> instance configured according to the given <a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a> instance. The population of Chromosomes will be created according to the setup of the sample <a class="el" href="classorg_1_1jgap_1_1_chromosome.html">Chromosome</a> in the <a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a> object, but the gene values (alleles) will be set to random legal values.</p>
<dl class="params"><dt>Parámetros</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_configuration</td><td>the current active <a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a> object </td></tr>
  </table>
  </dd>
</dl>
<dl class="section return"><dt>Devuelve</dt><dd>a newly constructed <a class="el" href="classorg_1_1jgap_1_1_genotype.html">Genotype</a> instance</dd></dl>
<dl class="exception"><dt>Excepciones</dt><dd>
  <table class="exception">
    <tr><td class="paramname">IllegalArgumentException</td><td>if the given <a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a> object is null </td></tr>
    <tr><td class="paramname"><a class="el" href="classorg_1_1jgap_1_1_invalid_configuration_exception.html">InvalidConfigurationException</a></td><td>if the given <a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a> instance is not in a valid state</td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Autor</dt><dd>Neil Rotstan </dd>
<dd>
Klaus Meffert </dd></dl>
<dl class="section since"><dt>Desde</dt><dd>2.3 </dd></dl>
<div class="fragment"><div class="line"><a name="l00323"></a><span class="lineno">  323</span>&#160;                                           {</div>
<div class="line"><a name="l00324"></a><span class="lineno">  324</span>&#160;    <span class="keywordflow">if</span> (a_configuration == null) {</div>
<div class="line"><a name="l00325"></a><span class="lineno">  325</span>&#160;      <span class="keywordflow">throw</span> <span class="keyword">new</span> IllegalArgumentException(</div>
<div class="line"><a name="l00326"></a><span class="lineno">  326</span>&#160;          <span class="stringliteral">&quot;The Configuration instance may not be null.&quot;</span>);</div>
<div class="line"><a name="l00327"></a><span class="lineno">  327</span>&#160;    }</div>
<div class="line"><a name="l00328"></a><span class="lineno">  328</span>&#160;    a_configuration.lockSettings();</div>
<div class="line"><a name="l00329"></a><span class="lineno">  329</span>&#160;    <span class="comment">// Create an array of chromosomes equal to the desired size in the</span></div>
<div class="line"><a name="l00330"></a><span class="lineno">  330</span>&#160;    <span class="comment">// active Configuration and then populate that array with Chromosome</span></div>
<div class="line"><a name="l00331"></a><span class="lineno">  331</span>&#160;    <span class="comment">// instances constructed according to the setup in the sample</span></div>
<div class="line"><a name="l00332"></a><span class="lineno">  332</span>&#160;    <span class="comment">// Chromosome, but with random gene values (alleles). The Chromosome</span></div>
<div class="line"><a name="l00333"></a><span class="lineno">  333</span>&#160;    <span class="comment">// class randomInitialChromosome() method will take care of that for</span></div>
<div class="line"><a name="l00334"></a><span class="lineno">  334</span>&#160;    <span class="comment">// us.</span></div>
<div class="line"><a name="l00335"></a><span class="lineno">  335</span>&#160;    <span class="comment">// ------------------------------------------------------------------</span></div>
<div class="line"><a name="l00336"></a><span class="lineno">  336</span>&#160;    <span class="keywordtype">int</span> populationSize = a_configuration.getPopulationSize();</div>
<div class="line"><a name="l00337"></a><span class="lineno">  337</span>&#160;    Population pop = <span class="keyword">new</span> Population(a_configuration, populationSize);</div>
<div class="line"><a name="l00338"></a><span class="lineno">  338</span>&#160;    <span class="comment">// Do randomized initialization.</span></div>
<div class="line"><a name="l00339"></a><span class="lineno">  339</span>&#160;    <span class="comment">// -----------------------------</span></div>
<div class="line"><a name="l00340"></a><span class="lineno">  340</span>&#160;    <a class="code" href="classorg_1_1jgap_1_1_genotype.html#a00b11af9a68e9d84bbc52afbf661607f">Genotype</a> result = <span class="keyword">new</span> <a class="code" href="classorg_1_1jgap_1_1_genotype.html#a00b11af9a68e9d84bbc52afbf661607f">Genotype</a>(a_configuration, pop);</div>
<div class="line"><a name="l00341"></a><span class="lineno">  341</span>&#160;    result.fillPopulation(populationSize);</div>
<div class="line"><a name="l00342"></a><span class="lineno">  342</span>&#160;    <span class="keywordflow">return</span> result;</div>
<div class="line"><a name="l00343"></a><span class="lineno">  343</span>&#160;  }</div>
<div class="ttc" id="classorg_1_1jgap_1_1_genotype_html_a00b11af9a68e9d84bbc52afbf661607f"><div class="ttname"><a href="classorg_1_1jgap_1_1_genotype.html#a00b11af9a68e9d84bbc52afbf661607f">org.jgap.Genotype.Genotype</a></div><div class="ttdeci">Genotype(Configuration a_configuration, IChromosome[] a_initialChromosomes)</div><div class="ttdef"><b>Definition:</b> Genotype.java:76</div></div>
</div><!-- fragment -->
</div>
</div>
<a class="anchor" id="a0d14c5400197e3c43f9d1f0666309a85"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.Genotype.run </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
</div><div class="memdoc">
<p>If used in a Thread: runs the evolution forever. You have to implement a listener to stop computation sometime. See <a class="el" href="namespaceexamples_1_1simple_boolean_threaded.html">examples.simpleBooleanThreaded</a> for a possible implementation of such a listener.</p>
<dl class="section author"><dt>Autor</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Desde</dt><dd>3.01 </dd></dl>
<div class="fragment"><div class="line"><a name="l00624"></a><span class="lineno">  624</span>&#160;                    {</div>
<div class="line"><a name="l00625"></a><span class="lineno">  625</span>&#160;    <span class="keywordflow">while</span> (!Thread.currentThread().interrupted()) {</div>
<div class="line"><a name="l00626"></a><span class="lineno">  626</span>&#160;      <span class="keywordflow">if</span>(m_useMonitor) {</div>
<div class="line"><a name="l00627"></a><span class="lineno">  627</span>&#160;       <a class="code" href="classorg_1_1jgap_1_1_genotype.html#a54a6791fbb6e13920e79f4e2dd719eb4">evolve</a>(m_monitor);</div>
<div class="line"><a name="l00628"></a><span class="lineno">  628</span>&#160;      }</div>
<div class="line"><a name="l00629"></a><span class="lineno">  629</span>&#160;      <span class="keywordflow">else</span> {</div>
<div class="line"><a name="l00630"></a><span class="lineno">  630</span>&#160;        <a class="code" href="classorg_1_1jgap_1_1_genotype.html#a54a6791fbb6e13920e79f4e2dd719eb4">evolve</a>();</div>
<div class="line"><a name="l00631"></a><span class="lineno">  631</span>&#160;      }</div>
<div class="line"><a name="l00632"></a><span class="lineno">  632</span>&#160;    }</div>
<div class="line"><a name="l00633"></a><span class="lineno">  633</span>&#160;  }</div>
<div class="ttc" id="classorg_1_1jgap_1_1_genotype_html_a54a6791fbb6e13920e79f4e2dd719eb4"><div class="ttname"><a href="classorg_1_1jgap_1_1_genotype.html#a54a6791fbb6e13920e79f4e2dd719eb4">org.jgap.Genotype.evolve</a></div><div class="ttdeci">synchronized void evolve()</div><div class="ttdef"><b>Definition:</b> Genotype.java:231</div></div>
</div><!-- fragment -->
</div>
</div>
<a class="anchor" id="af3f49ba90cc8515817366792b7bce320"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.Genotype.setMonitor </td>
          <td>(</td>
          <td class="paramtype">IEvolutionMonitor&#160;</td>
          <td class="paramname"><em>a_monitor</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
</div><div class="memdoc">
<p>Sets the evolution monitor to use, activate it via setUseMonitor(true). Only to be used when running <a class="el" href="classorg_1_1jgap_1_1_genotype.html">Genotype</a> as a thread.</p>
<dl class="params"><dt>Parámetros</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_monitor</td><td>the IEvolutionMonitor to use</td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Autor</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Desde</dt><dd>3.6 </dd></dl>
<div class="fragment"><div class="line"><a name="l00716"></a><span class="lineno">  716</span>&#160;                                                      {</div>
<div class="line"><a name="l00717"></a><span class="lineno">  717</span>&#160;    m_monitor = a_monitor;</div>
<div class="line"><a name="l00718"></a><span class="lineno">  718</span>&#160;  }</div>
</div><!-- fragment -->
</div>
</div>
<a class="anchor" id="ac56a4d112334db07ba333602c4a24706"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.Genotype.setPopulation </td>
          <td>(</td>
          <td class="paramtype"><a class="el" href="classorg_1_1jgap_1_1_population.html">Population</a>&#160;</td>
          <td class="paramname"><em>a_pop</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">protected</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<div class="fragment"><div class="line"><a name="l00570"></a><span class="lineno">  570</span>&#160;                                                 {</div>
<div class="line"><a name="l00571"></a><span class="lineno">  571</span>&#160;    m_population = a_pop;</div>
<div class="line"><a name="l00572"></a><span class="lineno">  572</span>&#160;  }</div>
</div><!-- fragment -->
</div>
</div>
<a class="anchor" id="a2f44a7709eca83cddd6d794ee59e663c"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">static void org.jgap.Genotype.setStaticConfiguration </td>
          <td>(</td>
          <td class="paramtype"><a class="el" href="classorg_1_1jgap_1_1_configuration.html">Configuration</a>&#160;</td>
          <td class="paramname"><em>a_configuration</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">static</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Sets the configuration to use with the Genetic Algorithm. </p>
<dl class="params"><dt>Parámetros</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_configuration</td><td>the configuration to use</td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Autor</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Desde</dt><dd>2.0 </dd></dl>
<div class="fragment"><div class="line"><a name="l00531"></a><span class="lineno">  531</span>&#160;                                                                           {</div>
<div class="line"><a name="l00532"></a><span class="lineno">  532</span>&#160;    m_staticConfiguration = a_configuration;</div>
<div class="line"><a name="l00533"></a><span class="lineno">  533</span>&#160;  }</div>
</div><!-- fragment -->
</div>
</div>
<a class="anchor" id="abba6c68b3e8388be6884a35f5600e8cb"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">void org.jgap.Genotype.setUseMonitor </td>
          <td>(</td>
          <td class="paramtype">boolean&#160;</td>
          <td class="paramname"><em>a_useMonitor</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
</div><div class="memdoc">
<p>Use an evolution monitor, only to be used when running <a class="el" href="classorg_1_1jgap_1_1_genotype.html">Genotype</a> as a thread. Otherwise use method <a class="el" href="classorg_1_1jgap_1_1_genotype.html#aa740dd753d31620166979199a38d5a50">evolve(IEvolutionMonitor)</a></p>
<dl class="params"><dt>Parámetros</dt><dd>
  <table class="params">
    <tr><td class="paramname">a_useMonitor</td><td>true: use evolution monitor, set it via setMonitor</td></tr>
  </table>
  </dd>
</dl>
<dl class="section author"><dt>Autor</dt><dd>Klaus Meffert </dd></dl>
<dl class="section since"><dt>Desde</dt><dd>3.6 </dd></dl>
<div class="fragment"><div class="line"><a name="l00703"></a><span class="lineno">  703</span>&#160;                                                  {</div>
<div class="line"><a name="l00704"></a><span class="lineno">  704</span>&#160;    m_useMonitor = a_useMonitor;</div>
<div class="line"><a name="l00705"></a><span class="lineno">  705</span>&#160;  }</div>
</div><!-- fragment -->
</div>
</div>
<a class="anchor" id="a658a541940e9be8ef474375091ff31ae"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">String org.jgap.Genotype.toString </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
</div><div class="memdoc">
<dl class="section return"><dt>Devuelve</dt><dd>string representation of this <a class="el" href="classorg_1_1jgap_1_1_genotype.html">Genotype</a> instance, useful for display or debug purposes</dd></dl>
<dl class="section author"><dt>Autor</dt><dd>Neil Rotstan </dd></dl>
<dl class="section since"><dt>Desde</dt><dd>1.0 </dd></dl>
<div class="fragment"><div class="line"><a name="l00292"></a><span class="lineno">  292</span>&#160;                           {</div>
<div class="line"><a name="l00293"></a><span class="lineno">  293</span>&#160;    StringBuffer buffer = <span class="keyword">new</span> StringBuffer();</div>
<div class="line"><a name="l00294"></a><span class="lineno">  294</span>&#160;    <span class="keywordflow">for</span> (<span class="keywordtype">int</span> i = 0; i &lt; <a class="code" href="classorg_1_1jgap_1_1_genotype.html#a8241b2d89d32b283abe5c9b42974d64d">getPopulation</a>().size(); i++) {</div>
<div class="line"><a name="l00295"></a><span class="lineno">  295</span>&#160;      buffer.append(<a class="code" href="classorg_1_1jgap_1_1_genotype.html#a8241b2d89d32b283abe5c9b42974d64d">getPopulation</a>().getChromosome(i).<a class="code" href="classorg_1_1jgap_1_1_genotype.html#a658a541940e9be8ef474375091ff31ae">toString</a>());</div>
<div class="line"><a name="l00296"></a><span class="lineno">  296</span>&#160;      buffer.append(<span class="stringliteral">&quot; [&quot;</span>);</div>
<div class="line"><a name="l00297"></a><span class="lineno">  297</span>&#160;      buffer.append(<a class="code" href="classorg_1_1jgap_1_1_genotype.html#a8241b2d89d32b283abe5c9b42974d64d">getPopulation</a>().getChromosome(i).getFitnessValueDirectly());</div>
<div class="line"><a name="l00298"></a><span class="lineno">  298</span>&#160;      buffer.append(<span class="stringliteral">&quot;]\n&quot;</span>);</div>
<div class="line"><a name="l00299"></a><span class="lineno">  299</span>&#160;    }</div>
<div class="line"><a name="l00300"></a><span class="lineno">  300</span>&#160;    <span class="keywordflow">return</span> buffer.toString();</div>
<div class="line"><a name="l00301"></a><span class="lineno">  301</span>&#160;  }</div>
<div class="ttc" id="classorg_1_1jgap_1_1_genotype_html_a8241b2d89d32b283abe5c9b42974d64d"><div class="ttname"><a href="classorg_1_1jgap_1_1_genotype.html#a8241b2d89d32b283abe5c9b42974d64d">org.jgap.Genotype.getPopulation</a></div><div class="ttdeci">Population getPopulation()</div><div class="ttdef"><b>Definition:</b> Genotype.java:167</div></div>
<div class="ttc" id="classorg_1_1jgap_1_1_genotype_html_a658a541940e9be8ef474375091ff31ae"><div class="ttname"><a href="classorg_1_1jgap_1_1_genotype.html#a658a541940e9be8ef474375091ff31ae">org.jgap.Genotype.toString</a></div><div class="ttdeci">String toString()</div><div class="ttdef"><b>Definition:</b> Genotype.java:292</div></div>
</div><!-- fragment -->
</div>
</div>
<hr/>La documentación para esta clase fue generada a partir del siguiente fichero:<ul>
<li>src/org/jgap/<a class="el" href="_genotype_8java.html">Genotype.java</a></li>
</ul>
</div><!-- contents -->
</div><!-- doc-content -->
<!-- start footer part -->
<div id="nav-path" class="navpath"><!-- id is needed for treeview function! -->
  <ul>
    <li class="navelem"><a class="el" href="namespaceorg.html">org</a></li><li class="navelem"><a class="el" href="namespaceorg_1_1jgap.html">jgap</a></li><li class="navelem"><a class="el" href="classorg_1_1jgap_1_1_genotype.html">Genotype</a></li>
    <li class="footer">Generado el Domingo, 24 de Noviembre de 2013 18:23:15 para Proyecto JGAP por
    <a href="http://www.doxygen.org/index.html">
    <img class="footer" src="doxygen.png" alt="doxygen"/></a> 1.8.5 </li>
  </ul>
</div>
</body>
</html>
