<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.5"/>
<title>Proyecto JGAP: Lista de tareas pendientes</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="navtree.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="resize.js"></script>
<script type="text/javascript" src="navtree.js"></script>
<script type="text/javascript">
  $(document).ready(initResizable);
  $(window).load(resizeHeight);
</script>
<link href="search/search.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="search/search.js"></script>
<script type="text/javascript">
  $(document).ready(function() { searchBox.OnSelectItem(0); });
</script>
<link href="doxygen.css" rel="stylesheet" type="text/css" />
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<div id="titlearea">
<table cellspacing="0" cellpadding="0">
 <tbody>
 <tr style="height: 56px;">
  <td style="padding-left: 0.5em;">
   <div id="projectname">Proyecto JGAP
   &#160;<span id="projectnumber">1.0</span>
   </div>
   <div id="projectbrief">Proyecto JGAP</div>
  </td>
 </tr>
 </tbody>
</table>
</div>
<!-- end header part -->
<!-- Generado por Doxygen 1.8.5 -->
<script type="text/javascript">
var searchBox = new SearchBox("searchBox", "search",false,'Buscar');
</script>
  <div id="navrow1" class="tabs">
    <ul class="tablist">
      <li><a href="index.html"><span>PÃ¡gina&#160;principal</span></a></li>
      <li class="current"><a href="pages.html"><span>PÃ¡ginas&#160;relacionadas</span></a></li>
      <li><a href="namespaces.html"><span>Paquetes</span></a></li>
      <li><a href="annotated.html"><span>Clases</span></a></li>
      <li><a href="files.html"><span>Archivos</span></a></li>
      <li>
        <div id="MSearchBox" class="MSearchBoxInactive">
        <span class="left">
          <img id="MSearchSelect" src="search/mag_sel.png"
               onmouseover="return searchBox.OnSearchSelectShow()"
               onmouseout="return searchBox.OnSearchSelectHide()"
               alt=""/>
          <input type="text" id="MSearchField" value="Buscar" accesskey="S"
               onfocus="searchBox.OnSearchFieldFocus(true)" 
               onblur="searchBox.OnSearchFieldFocus(false)" 
               onkeyup="searchBox.OnSearchFieldChange(event)"/>
          </span><span class="right">
            <a id="MSearchClose" href="javascript:searchBox.CloseResultsWindow()"><img id="MSearchCloseImg" border="0" src="search/close.png" alt=""/></a>
          </span>
        </div>
      </li>
    </ul>
  </div>
</div><!-- top -->
<div id="side-nav" class="ui-resizable side-nav-resizable">
  <div id="nav-tree">
    <div id="nav-tree-contents">
      <div id="nav-sync" class="sync"></div>
    </div>
  </div>
  <div id="splitbar" style="-moz-user-select:none;" 
       class="ui-resizable-handle">
  </div>
</div>
<script type="text/javascript">
$(document).ready(function(){initNavTree('todo.html','');});
</script>
<div id="doc-content">
<!-- window showing the filter options -->
<div id="MSearchSelectWindow"
     onmouseover="return searchBox.OnSearchSelectShow()"
     onmouseout="return searchBox.OnSearchSelectHide()"
     onkeydown="return searchBox.OnSearchSelectKey(event)">
<a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(0)"><span class="SelectionMark">&#160;</span>Todo</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(1)"><span class="SelectionMark">&#160;</span>Clases</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(2)"><span class="SelectionMark">&#160;</span>Namespaces</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(3)"><span class="SelectionMark">&#160;</span>Archivos</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(4)"><span class="SelectionMark">&#160;</span>Funciones</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(5)"><span class="SelectionMark">&#160;</span>Variables</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(6)"><span class="SelectionMark">&#160;</span>PÃ¡ginas</a></div>

<!-- iframe showing the search results (closed by default) -->
<div id="MSearchResultsWindow">
<iframe src="javascript:void(0)" frameborder="0" 
        name="MSearchResults" id="MSearchResults">
</iframe>
</div>

<div class="header">
  <div class="headertitle">
<div class="title">Lista de tareas pendientes </div>  </div>
</div><!--header-->
<div class="contents">
<div class="textblock"><dl class="reflist">
<dt><a class="anchor" id="_todo000001"></a>Miembro <a class="el" href="classexamples_1_1audit_1_1_coins_example.html#ac0a78806e49f802c46d856b1b666d815">examples.audit.CoinsExample.makeChangeForAmount</a>  (int a_targetChangeAmount)</dt>
<dd>class Evaluator: input:<ul>
<li>PermutingConfiguration</li>
<li>Number of evaluation runs pers config (to turn off randomness as much as possible)</li>
<li>output facility (data container)</li>
<li>optional: event subscribers output:</li>
<li>averaged curve of fitness value thru all generations</li>
<li>best fitness value accomplished</li>
<li>average number of performance improvements for all generations  </li>
</ul>
</dd>
<dt><a class="anchor" id="_todo000003"></a>Miembro <a class="el" href="classexamples_1_1gp_1_1anttrail_1_1_ant_map.html#a659ac2b9f81a55d17fb2e236574c8c70">examples.gp.anttrail.AntMap.AntMap</a>  (final int[][] a_map, int a_maxMoves)</dt>
<dd>speedup possible by using string?  </dd>
<dt><a class="anchor" id="_todo000004"></a>Miembro <a class="el" href="classexamples_1_1gp_1_1_fibonacci.html#a09f676281515cca71bae669293152c09">examples.gp.Fibonacci.create</a>  ()</dt>
<dd>allow to optionally preset a static program in each chromosome  </dd>
<dt><a class="anchor" id="_todo000007"></a>Miembro <a class="el" href="classexamples_1_1gp_1_1island_1_1_island_g_p_example.html#a830119e3f370d14fb0be6187f1d9aa28">examples.gp.island.IslandGPExample.IslandGPExample</a>  ()</dt>
<dd><p class="startdd">Island config. einbauen, so dass Island model automatisch ausgeführt wird </p>
<p>impl. MergeEvent, e.g.: merge islands when one island stucks </p>
<p class="enddd">jeweils beste lösung geht öfters verloren! Beeinflusst ein Island das andere oder ist die Lösung in einem Island nicht beständig?  </p>
</dd>
<dt><a class="anchor" id="_todo000010"></a>Miembro <a class="el" href="classexamples_1_1gp_1_1tictactoe_1_1_evaluate_board.html#abece9125d9ee8fb48cf6aa5a96af7c0a">examples.gp.tictactoe.EvaluateBoard.execute_void</a>  (ProgramChromosome c, int n, Object[] args)</dt>
<dd>support other types than integer  </dd>
<dt><a class="anchor" id="_todo000011"></a>Miembro <a class="el" href="classexamples_1_1gp_1_1tictactoe_1_1_single_node_validator.html#a32e7b83ad42a193123dbe4dbf9aefc25">examples.gp.tictactoe.SingleNodeValidator.isAllowed</a>  (int a_chromIndex, ProgramChromosome a_pc, CommandGene[] a_functionSet, CommandGene a_function, Class a_returnType, int a_subReturnTyp, int m_index)</dt>
<dd>provide convenient method setNonConsecutive(boolean) in CommandGene  </dd>
<dt><a class="anchor" id="_todo000017"></a>Miembro <a class="el" href="classexamples_1_1gp_1_1tictactoe_1_1_tic_tac_toe_main.html#a477f163a17746606ab745ce91e9bde7f">examples.gp.tictactoe.TicTacToeMain.create</a>  (GPConfiguration a_conf, int a_color, GPGenotype a_other, int a_otherColor)</dt>
<dd>evaluate board to matrix  </dd>
<dt><a class="anchor" id="_todo000020"></a>Miembro <a class="el" href="classexamples_1_1island_1_1_island_example.html#aad36c7af08e927224e83d83a04b104e1">examples.island.IslandExample.start</a>  ()</dt>
<dd></dd>
<dt><a class="anchor" id="_todo000021"></a>Miembro <a class="el" href="classexamples_1_1monalisa_1_1core_1_1_alpha_mutation_operator.html#ae1fcb87d1d1a9d22a86b7f4bb5593c60">examples.monalisa.core.AlphaMutationOperator.operate</a>  (final Population a_population, final List a_candidateChromosomes)</dt>
<dd>move to base class, refactor  </dd>
<dt><a class="anchor" id="_todo000022"></a>Miembro <a class="el" href="classexamples_1_1monalisa_1_1core_1_1_alpha_off_mutation_operator.html#a68240b26bc561f24ab1251cc97ae9252">examples.monalisa.core.AlphaOffMutationOperator.operate</a>  (final Population a_population, final List a_candidateChromosomes)</dt>
<dd>move to base class, refactor  </dd>
<dt><a class="anchor" id="_todo000023"></a>Miembro <a class="el" href="classexamples_1_1monalisa_1_1core_1_1_color_mutation_operator.html#a1143ed54d24aa0d922111a8bede53f0d">examples.monalisa.core.ColorMutationOperator.operate</a>  (final Population a_population, final List a_candidateChromosomes)</dt>
<dd>move to base class, refactor  </dd>
<dt><a class="anchor" id="_todo000024"></a>Miembro <a class="el" href="classexamples_1_1monalisa_1_1core_1_1_point_mutation_operator.html#a77dc2c33c39c88ed99f6300dd8022125">examples.monalisa.core.PointMutationOperator.operate</a>  (final Population a_population, final List a_candidateChromosomes)</dt>
<dd>move to base class, refactor  </dd>
<dt><a class="anchor" id="_todo000025"></a>Miembro <a class="el" href="classexamples_1_1monalisa_1_1core_1_1_polygon_mutation_operator.html#a007895c5c90286148d47ef8d1cd1ce93">examples.monalisa.core.PolygonMutationOperator.operate</a>  (final Population a_population, final List a_candidateChromosomes)</dt>
<dd>move to base class, refactor  </dd>
<dt><a class="anchor" id="_todo000026"></a>Miembro <a class="el" href="classexamples_1_1monalisa_1_1core_1_1_single_mutation_operator.html#a07db0eb33cb05fa3407e1b560e0a7a9a">examples.monalisa.core.SingleMutationOperator.operate</a>  (final Population a_population, final List a_candidateChromosomes)</dt>
<dd>move to base class, refactor  </dd>
<dt><a class="anchor" id="_todo000027"></a>Miembro <a class="el" href="classexamples_1_1simple_boolean_threaded_1_1_simple_example.html#ab55245566b467d39fbc8098617ba5cbe">examples.simpleBooleanThreaded.SimpleExample.main</a>  (String[] args)</dt>
<dd>evaluate monitor data  </dd>
<dt><a class="anchor" id="_todo000029"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1audit_1_1_evaluator.html#a68331d6b5db79708b87b72d5077a0e5e">org.jgap.audit.Evaluator.calcAvgFitnessImpr</a>  (int a_permutation)</dt>
<dd><p class="startdd">implement </p>
<p>implement </p>
<p class="enddd">is this method used resp. contained in calcPerformance?  </p>
</dd>
<dt><a class="anchor" id="_todo000032"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1audit_1_1_evaluator.html#a48f84bb9929cec12795c98bf1185ac68">org.jgap.audit.Evaluator.storePopulation</a>  (int a_permutation, int a_run, Population a_pop)</dt>
<dd>implement  </dd>
<dt><a class="anchor" id="_todo000033"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1audit_1_1_evolution_evaluator.html#a411341e5eb1abbec8e0995d8fb49727a">org.jgap.audit.EvolutionEvaluator.evaluate</a>  ()</dt>
<dd><p class="startdd">use chromIDs and chromTrack </p>
<p class="enddd"></p>
</dd>
<dt><a class="anchor" id="_todo000035"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1audit_1_1_permuting_configuration.html#a541cb4586f480c4991369f04753a2be8">org.jgap.audit.PermutingConfiguration.next</a>  ()</dt>
<dd><p class="startdd">make permutation below computed dynamic and not static </p>
<p>introduce new parameters: populationSize, setPreservFittestIndividual, MAX_ALLOWED_EVOLUTIONS </p>
<p>allow for "false" </p>
<p class="enddd">BulkFitnessOffsetRemover vs. FitnessFunction  </p>
</dd>
<dt><a class="anchor" id="_todo000039"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1_base_chromosome.html#a30276dc8afb0bd632c288bf7d2e18b55">org.jgap.BaseChromosome.setValueFromPersistentRepresentation</a>  (String a_representation)</dt>
<dd><p class="startdd">we can do this faster! </p>
<p class="enddd">we can do this faster!  </p>
</dd>
<dt><a class="anchor" id="_todo000041"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1_base_gene.html#a40f8f3d249145ad028892cd706ba9dae">org.jgap.BaseGene.newGene</a>  ()</dt>
<dd>clone app.data  </dd>
<dt><a class="anchor" id="_todo000184"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1_base_gene_test.html#a8e859f6d03d8eb3ce5dfe926d11eed92">org.jgap.BaseGeneTest.testEquals_3</a>  ()</dt>
<dd>use other than JGAPFactory to be able to receive a null CompareToHandler for the application data object  </dd>
<dt><a class="anchor" id="_todo000185"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1_base_gene_test.html#a00bdb1c53bcf6d47c9ad16fec5a52de3">org.jgap.BaseGeneTest.testSetApplicationData_1</a>  ()</dt>
<dd>find the one chromosome  </dd>
<dt><a class="anchor" id="_todo000186"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1_base_gene_test.html#ab2cd872d724507f006962aecc05830f6">org.jgap.BaseGeneTest.testSetApplicationData_2</a>  ()</dt>
<dd>find the one chromosome  </dd>
<dt><a class="anchor" id="_todo000043"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1_base_genetic_operator.html#a4db6fa1a99177114c0f19f02cd4a87ec">org.jgap.BaseGeneticOperator.equals</a>  (final Object a_other)</dt>
<dd>also compare Configuration?  </dd>
<dt><a class="anchor" id="_todo000042"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1_base_genetic_operator.html#addbc6b98f572331d47397227b7a49f6a">org.jgap.BaseGeneticOperator.getConfiguration</a>  ()</dt>
<dd>make returned object immutable  </dd>
<dt><a class="anchor" id="_todo000046"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1_breeder_base.html#ac2baa98b33eb5a909ffddc874598c1d2">org.jgap.BreederBase.applyGeneticOperators</a>  (Configuration a_config, Population a_pop)</dt>
<dd>utilize jobs: integrate job into GeneticOperator  </dd>
<dt><a class="anchor" id="_todo000044"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1_breeder_base.html#a3d2ad9ec1c90d78a5598a4dec1400960">org.jgap.BreederBase.applyNaturalSelectors</a>  (Configuration a_config, Population a_pop, boolean a_processBeforeGeneticOperators)</dt>
<dd><p class="startdd">optionally use working pool </p>
<p class="enddd">utilize jobs: integrate job into NaturalSelector!  </p>
</dd>
<dt><a class="anchor" id="_todo000049"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1_chromosome.html#a9f590d55a34d8b99c2a9a1b4aedd568b">org.jgap.Chromosome.clone</a>  ()</dt>
<dd>once output a warning: allele should be cloneable!  </dd>
<dt><a class="anchor" id="_todo000052"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1_chromosome.html#a2be0507944e2ca98bef8372402b33e13">org.jgap.Chromosome.compareTo</a>  (Object other)</dt>
<dd>improve  </dd>
<dt><a class="anchor" id="_todo000050"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1_chromosome.html#a6b6fb5c7c5b532813a4661de5dbd9ab3">org.jgap.Chromosome.randomInitialChromosome</a>  (Configuration a_configuration)</dt>
<dd><p class="startdd">what about Gene's energy? </p>
<p class="enddd">what about Gene's energy?  </p>
</dd>
<dt><a class="anchor" id="_todo000053"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1_configuration.html#a453c6b764b4f017c058632bc22e5a915">org.jgap.Configuration.doClone</a>  (Object a_objToClone)</dt>
<dd>try cloning in a standard way  </dd>
<dt><a class="anchor" id="_todo000060"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1distr_1_1_breeder.html#ab3ff454f082f23bb5885ea8e0aa2ed89">org.jgap.distr.Breeder.informParent</a>  ()</dt>
<dd>implement  </dd>
<dt><a class="anchor" id="_todo000061"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1distr_1_1_breeder.html#ad371c6cce93099457783ac1fe6da74ca">org.jgap.distr.Breeder.stop</a>  ()</dt>
<dd>implement  </dd>
<dt><a class="anchor" id="_todo000064"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1distr_1_1grid_1_1_dummy_grid_client_mediator.html#a7212ea568d78375a552bc599e62e4648">org.jgap.distr.grid.DummyGridClientMediator.getGridMessage</a>  (MessageContext a_context, String a_datetime, int a_timeoutSeconds, int a_waitTimeSeconds, boolean a_removeRequest)</dt>
<dd>fallback solution  </dd>
<dt><a class="anchor" id="_todo000063"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1distr_1_1grid_1_1_dummy_grid_client_mediator.html#a080ac41bf4278214c07055ebcc9060f8">org.jgap.distr.grid.DummyGridClientMediator.startClient</a>  ()</dt>
<dd>allow asynchronous wait for server (check for "java.net.ConnectException: Connection refused: connect")  </dd>
<dt><a class="anchor" id="_todo000065"></a>Miembro <a class="el" href="interfaceorg_1_1jgap_1_1distr_1_1grid_1_1gp_1_1_i_grid_configuration_g_p.html#aa7705e8e26bdc18c2c0e5b69e8ac818a">org.jgap.distr.grid.gp.IGridConfigurationGP.setTypes</a>  (Class[] a_types)</dt>
<dd>move the following to GPConfiguration, elegantly  </dd>
<dt><a class="anchor" id="_todo000077"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1distr_1_1grid_1_1gp_1_1_j_g_a_p_client_g_p.html#acfb2127d50b79830207941705b3f268a">org.jgap.distr.grid.gp.JGAPClientGP.JGAPClientGP</a>  (IGridClientMediator a_gcmed, GridNodeClientConfig a_gridconfig, String a_clientClassName, boolean a_WANMode, boolean a_receiveOnly, boolean a_list, boolean a_no_comm, boolean a_no_evolution, boolean a_endless, int a_max_fetch_results)</dt>
<dd>ab 2. zyklus ist pop.grösse nur 1 !  </dd>
<dt><a class="anchor" id="_todo000075"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1distr_1_1grid_1_1gp_1_1_j_g_a_p_client_g_p.html#a0535356c14866325a1f7df6faa19db92">org.jgap.distr.grid.gp.JGAPClientGP.MODULE_CS</a>  </dt>
<dd>store version in external file  </dd>
<dt><a class="anchor" id="_todo000076"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1distr_1_1grid_1_1gp_1_1_j_g_a_p_client_g_p.html#a6424bb7eb2e4963a50cba03b8e28d241">org.jgap.distr.grid.gp.JGAPClientGP.MODULE_SC</a>  </dt>
<dd>das ist nicht module, sondern sender-receiver  </dd>
<dt><a class="anchor" id="_todo000093"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1distr_1_1grid_1_1gp_1_1_j_g_a_p_client_g_p.html#adf5018d320d27b924e0f16a2ac48fea9">org.jgap.distr.grid.gp.JGAPClientGP.presetPopulation</a>  (JGAPRequestGP a_workRequest)</dt>
<dd>remove results[index]  </dd>
<dt><a class="anchor" id="_todo000080"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1distr_1_1grid_1_1gp_1_1_j_g_a_p_client_g_p.html#a7a6ca719b3c2154e9f85d3d09228e523">org.jgap.distr.grid.gp.JGAPClientGP.receiveWorkResults</a>  (JGAPRequestGP[] workList)</dt>
<dd><p class="startdd">later: SC </p>
<p>sort results according to post date, the oldest first </p>
<p>config.params wie popsite, evol.anz dazu </p>
<p>allow fitness 0.0 </p>
<p class="enddd">do this here explicitely and not in receiveWorkResult  </p>
</dd>
<dt><a class="anchor" id="_todo000088"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1distr_1_1grid_1_1gp_1_1_j_g_a_p_client_g_p.html#a4151e753398914b0d1d945464fa62e2d">org.jgap.distr.grid.gp.JGAPClientGP.resultReceived</a>  (IGPProgram a_fittest)</dt>
<dd><p class="startdd">jeden Worker einer von n (rein logischen) Gruppen zuteilen. Pro logischer Gruppe top n Ergebnisse halten </p>
<p>make configurable </p>
<p>compare with toStringNorm(0), use remove(int) instead of remove(Object) </p>
<p>ist nur test! </p>
<p class="enddd">skip unnecessary data, inject it after reload  </p>
</dd>
<dt><a class="anchor" id="_todo000078"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1distr_1_1grid_1_1gp_1_1_j_g_a_p_client_g_p.html#a256acdc898522793fa73e386b9cbf4e9">org.jgap.distr.grid.gp.JGAPClientGP.sendWorkRequests</a>  (int a_evolutionIndex, IClientEvolveStrategyGP evolver, IRequestSplitStrategyGP splitter, IClientFeedbackGP feedback)</dt>
<dd>merge previous results in req.getPopulation()  </dd>
<dt><a class="anchor" id="_todo000079"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1distr_1_1grid_1_1gp_1_1_j_g_a_p_client_g_p.html#ada87d248b0be00887061959c808b5ff7">org.jgap.distr.grid.gp.JGAPClientGP.sendWorkRequests</a>  (JGAPRequestGP[] a_workList)</dt>
<dd>hier ist fehler aufgetreten!  </dd>
<dt><a class="anchor" id="_todo000095"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1distr_1_1grid_1_1gp_1_1_j_g_a_p_worker_g_p.html#aa04fd0437862a49fe5fce45084c0590f">org.jgap.distr.grid.gp.JGAPWorkerGP.doWork</a>  (WorkRequest work, String workDir)</dt>
<dd>set gridworkerfeedback in class GridWorker  </dd>
<dt><a class="anchor" id="_todo000096"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1distr_1_1grid_1_1gp_1_1_j_g_a_p_workers_g_p.html#a45d28e995ace9be9709cf592c69d5b39">org.jgap.distr.grid.gp.JGAPWorkersGP.getNeededFiles</a>  (GridNodeWorkerConfig a_config)</dt>
<dd>implement  </dd>
<dt><a class="anchor" id="_todo000098"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1distr_1_1grid_1_1_j_g_a_p_client_handler_thread.html#a5513a2e631ddea023820d49fc322a781">org.jgap.distr.grid.JGAPClientHandlerThread.handleMsg</a>  (GridMessage msg)</dt>
<dd>consider 4GB limit  </dd>
<dt><a class="anchor" id="_todo000099"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1distr_1_1grid_1_1_j_g_a_p_worker.html#ae03c268f54333b88162959811590558d">org.jgap.distr.grid.JGAPWorker.doWork</a>  (WorkRequest work, String workDir)</dt>
<dd>set gridworkerfeedback in class GridWorker  </dd>
<dt><a class="anchor" id="_todo000100"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1distr_1_1grid_1_1_j_g_a_p_workers.html#a644cc69c70eb0ed5553eaa60e3dadbe7">org.jgap.distr.grid.JGAPWorkers.getNeededFiles</a>  (GridNodeWorkerConfig a_config)</dt>
<dd></dd>
<dt><a class="anchor" id="_todo000101"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1distr_1_1grid_1_1_request_split_strategy.html#afa4582eef1f6ae141ff15d706cf7ec68">org.jgap.distr.grid.RequestSplitStrategy.split</a>  (JGAPRequest a_request)</dt>
<dd>support cloning of m_config  </dd>
<dt><a class="anchor" id="_todo000102"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1distr_1_1grid_1_1util_1_1_grid_kit.html#a7ed064739e97dbc91ad4cd11ac736864">org.jgap.distr.grid.util.GridKit.isUpdateAvailable</a>  (String BASE_URL, String a_moduleName, String currentVersion)</dt>
<dd>download a versions.txt file and obtain version per project  </dd>
<dt><a class="anchor" id="_todo000103"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1distr_1_1grid_1_1util_1_1_grid_kit.html#a61d160ee177cb0b07e5e5fda371ad48f">org.jgap.distr.grid.util.GridKit.updateModule</a>  (String a_filename, String a_workDir, String a_libDir)</dt>
<dd>write log: delete manually  </dd>
<dt><a class="anchor" id="_todo000104"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1distr_1_1grid_1_1util_1_1_grid_kit.html#afa78889ffda168512995505a1ded1b3b">org.jgap.distr.grid.util.GridKit.updateModuleLibrary</a>  (String BASE_URL, String a_moduleName, String a_libDir, String a_workDir)</dt>
<dd><p class="startdd">zuordnung irgendwo abspeichern </p>
<p>zuordnung irgendwo abspeichern </p>
<p class="enddd">zuordnung irgendwo abspeichern  </p>
</dd>
<dt><a class="anchor" id="_todo000109"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1distr_1_1_master.html#aef7c64089218808c9b7f0893e31fffbe">org.jgap.distr.Master.sendToWorker</a>  (final IWorker a_worker, final WorkerCommand a_command)</dt>
<dd>implement  </dd>
<dt><a class="anchor" id="_todo000108"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1distr_1_1_master.html#aec0e9db5db3316ec9ee903c3892f5796">org.jgap.distr.Master.stop</a>  ()</dt>
<dd>notify all workers to stop working??? No, better would be: next time master is available it can receive old results from workers. So, the workers need to store them for some time, until the master is able to receive the worker results.  </dd>
<dt><a class="anchor" id="_todo000113"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1distr_1_1_worker.html#a405f281453fcdd0046a370a001af92ef">org.jgap.distr.Worker.getStatus</a>  ()</dt>
<dd>implement: idle starting receiving task working sending result stopping stopped  </dd>
<dt><a class="anchor" id="_todo000114"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1distr_1_1_worker.html#aea81d8531d56e4035cd94c5278e5c604">org.jgap.distr.Worker.pause</a>  ()</dt>
<dd>implement: able to pause resp. in work?  </dd>
<dt><a class="anchor" id="_todo000116"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1distr_1_1_worker.html#afc27690dd01d55455271aba6dc32b434">org.jgap.distr.Worker.resume</a>  ()</dt>
<dd>implement: able to resum resp. paused?  </dd>
<dt><a class="anchor" id="_todo000111"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1distr_1_1_worker.html#aee8f7803e2655bdfb3aceff9a3f92eb7">org.jgap.distr.Worker.sendCommand</a>  (final WorkerCommand a_command)</dt>
<dd><p class="startdd">this should be moved to a thread </p>
<p class="enddd">implement: currently working? if yes, add to queue (if queue not full) if no: start work  </p>
</dd>
<dt><a class="anchor" id="_todo000115"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1distr_1_1_worker.html#a7ecb6526e7b056129e7f4f8a584865de">org.jgap.distr.Worker.stop</a>  ()</dt>
<dd>implement: able to stop resp. in work?  </dd>
<dt><a class="anchor" id="_todo000117"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1distr_1_1_worker_command.html#a9b2e86e89542048dc457e6d0a7bd85bd">org.jgap.distr.WorkerCommand.execute</a>  (final Object a_parameters)</dt>
<dd>implement  </dd>
<dt><a class="anchor" id="_todo000118"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1_genotype.html#a6c14c3a36aa8e4291840f0b9c0facb35">org.jgap.Genotype.applyNaturalSelectors</a>  (boolean a_processBeforeGeneticOperators)</dt>
<dd>optionally use working pool  </dd>
<dt><a class="anchor" id="_todo000119"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1_genotype.html#af5e224c5b7c05dbe02b3324d5ec16f2a">org.jgap.Genotype.keepPopSizeConstant</a>  (Population a_pop, int a_maxSize)</dt>
<dd>use StandardPostSelector instead?  </dd>
<dt><a class="anchor" id="_todo000120"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1_genotype.html#aa2b0341edc420ad61e797508b8bf4dd5">org.jgap.Genotype.mergeResults</a>  (IPopulationMerger a_merger, EvolveResult[] a_results)</dt>
<dd>use/enhance IPopulationMerger  </dd>
<dt><a class="anchor" id="_todo000187"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1_genotype_test.html#aa0d1c56bf7ef0e020f3d979f784dc5d3">org.jgap.GenotypeTest.testIsSerializable_0</a>  ()</dt>
<dd>following will be obsolete  </dd>
<dt><a class="anchor" id="_todo000121"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1gp_1_1_command_gene.html#a7d5b399cad245a0276bd0b62c9e5a4a8">org.jgap.gp.CommandGene.compareTo</a>  (Object a_other)</dt>
<dd>do it more precisely  </dd>
<dt><a class="anchor" id="_todo000122"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1gp_1_1_command_gene.html#a161c170b4493de85203147a84ac13a3b">org.jgap.gp.CommandGene.setValueFromPersistentRepresentation</a>  (final String a_representation)</dt>
<dd>finish  </dd>
<dt><a class="anchor" id="_todo000188"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1gp_1_1_command_gene_test.html#a670aa7b5c2021f31a719ea32218d6671">org.jgap.gp.CommandGeneTest.testEquals_3</a>  ()</dt>
<dd>use other than JGAPFactory to be able to receive a null CompareToHandler for the application data object  </dd>
<dt><a class="anchor" id="_todo000123"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1gp_1_1function_1_1_a_d_f.html#a5e377b15d81ca7ceec156614480980d7">org.jgap.gp.function.ADF.isValid</a>  (ProgramChromosome a_chrom)</dt>
<dd>enhance  </dd>
<dt><a class="anchor" id="_todo000124"></a>Miembro <a class="el" href="enumorg_1_1jgap_1_1gp_1_1function_1_1_count_matrix_1_1_count_type.html#a7ab04ee65255adb71c8f6a47b727b775">org.jgap.gp.function.CountMatrix.CountType.COLUMN</a>  </dt>
<dd>count over whole matrix  </dd>
<dt><a class="anchor" id="_todo000125"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1gp_1_1function_1_1_exchange_memory.html#a3e44871ecef23f1319115126c507492b">org.jgap.gp.function.ExchangeMemory.isAffectGlobalState</a>  ()</dt>
<dd>subclass from new abstract class MemoryCommand?  </dd>
<dt><a class="anchor" id="_todo000126"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1gp_1_1function_1_1_for_x_loop.html#ae448f21b7912eec72d496b62669993e6">org.jgap.gp.function.ForXLoop.execute_void</a>  (ProgramChromosome c, int n, Object[] args)</dt>
<dd>only consider variables appearing before FORX in the program tree  </dd>
<dt><a class="anchor" id="_todo000127"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1gp_1_1function_1_1_if.html#a3b36ce8fb25456894c4bb2d136bfc5ef">org.jgap.gp.function.If.execute_void</a>  (ProgramChromosome c, int n, Object[] args)</dt>
<dd>add option for type of first child to constructor  </dd>
<dt><a class="anchor" id="_todo000128"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1gp_1_1function_1_1_if_dyn.html#ac7b9aee1521e3c567cfb36655376faf6">org.jgap.gp.function.IfDyn.execute_void</a>  (ProgramChromosome c, int n, Object[] args)</dt>
<dd>add option for type of first child to constructor  </dd>
<dt><a class="anchor" id="_todo000129"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1gp_1_1function_1_1_push.html#a44b325b83a314c139630027e8664eb98">org.jgap.gp.function.Push.isAffectGlobalState</a>  ()</dt>
<dd>use this information  </dd>
<dt><a class="anchor" id="_todo000130"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1gp_1_1function_1_1_push.html#a6fc7b172011e49476bbf72e88f0512a6">org.jgap.gp.function.Push.isValid</a>  (ProgramChromosome a_program)</dt>
<dd>consider n (execute_int...)  </dd>
<dt><a class="anchor" id="_todo000133"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1gp_1_1function_1_1_sub_program.html#a9ef69a70d4ce63696a5825e6f1bbaae8">org.jgap.gp.function.SubProgram.applyMutation</a>  ()</dt>
<dd>use dynamizeArity instead!  </dd>
<dt><a class="anchor" id="_todo000131"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1gp_1_1function_1_1_sub_program.html#a310c68bd816320dab6cf5c797ef398dc">org.jgap.gp.function.SubProgram.execute_int</a>  (ProgramChromosome c, int n, Object[] args)</dt>
<dd>evaluate m_types  </dd>
<dt><a class="anchor" id="_todo000132"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1gp_1_1function_1_1_sub_program.html#a6d8e0b323c21b311572bde0fcb8ea3e3">org.jgap.gp.function.SubProgram.execute_void</a>  (ProgramChromosome c, int n, Object[] args)</dt>
<dd>evaluate m_types  </dd>
<dt><a class="anchor" id="_todo000134"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1gp_1_1function_1_1_transfer_memory.html#a3eb497c878c705b6a8006d87b5d1db70">org.jgap.gp.function.TransferMemory.isAffectGlobalState</a>  ()</dt>
<dd>subclass from new abstract class MemoryCommand?  </dd>
<dt><a class="anchor" id="_todo000136"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_branch_typing_cross.html#af4ff7487c5e532031d260dce1b66d74b">org.jgap.gp.impl.BranchTypingCross.doCross</a>  (ProgramChromosome a_c0, ProgramChromosome a_c1)</dt>
<dd>solve in general  </dd>
<dt><a class="anchor" id="_todo000135"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_branch_typing_cross.html#ac9c1c1bf7edeadd26dccbcdd9d33ffc3">org.jgap.gp.impl.BranchTypingCross.operate</a>  (final IGPProgram a_i1, final IGPProgram a_i2)</dt>
<dd>try to ensure uniqueness for unique commands: after selecting first node, check if there is a unique node in the sub tree. If so, check if it appears in the sub tree of the second node.  </dd>
<dt><a class="anchor" id="_todo000137"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_default_population_creator.html#af6c38314bce32cc040b6beaa0160521f">org.jgap.gp.impl.DefaultPopulationCreator.initialize</a>  (GPPopulation a_pop, Class[] a_types, Class[][] a_argTypes, CommandGene[][] a_nodeSets, int[] a_minDepths, int[] a_maxDepths, int a_maxNodes, boolean[] a_fullModeAllowed)</dt>
<dd>pass params (e.g. advanced yes/no for Robocode)  </dd>
<dt><a class="anchor" id="_todo000138"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_fitness_proportionate_selection.html#a1bba66dc3595d954f39bd4121bc1fa7f">org.jgap.gp.impl.FitnessProportionateSelection.select</a>  (GPGenotype a_genotype)</dt>
<dd>implement for deltaMode  </dd>
<dt><a class="anchor" id="_todo000141"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_configuration.html#a919b59fd6a2dd43716d28f661ccaf924">org.jgap.gp.impl.GPConfiguration.getCrossoverProb</a>  ()</dt>
<dd>implement something like that  </dd>
<dt><a class="anchor" id="_todo000140"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_configuration.html#a97498de731029f37e23ac9ff7bab3de2">org.jgap.gp.impl.GPConfiguration.init</a>  (boolean a_fullInit)</dt>
<dd>make reusable in class Configuration and reuse here from Configuration  </dd>
<dt><a class="anchor" id="_todo000144"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype.html#aa465c7cbfb45d77644b3d195782a12c9">org.jgap.gp.impl.GPGenotype.calcFitness</a>  ()</dt>
<dd><p class="startdd">get information from fitness function how calculation happened. In case of Robocode: return the robot competed against, in case the -enemies option was used without -battleAll </p>
<p>do something similar here as with Genotype.preserveFittestChromosome </p>
<p>introduce global value object to be passed to the listener </p>
<p class="enddd">should not happen but does with ensureUniqueness(..)  </p>
</dd>
<dt><a class="anchor" id="_todo000152"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype.html#aa3bf0b50a73684ff3fe1177017cf9bb8">org.jgap.gp.impl.GPGenotype.checkErroneousProg</a>  (IGPProgram a_prog, String s, boolean a_clearFitness, boolean a_active)</dt>
<dd>impl. cache  </dd>
<dt><a class="anchor" id="_todo000148"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype.html#ad495c700786596290a63e3d88a902766">org.jgap.gp.impl.GPGenotype.evolve</a>  ()</dt>
<dd><p class="startdd">use program creator in case such is registered and returns a non-null program </p>
<p class="enddd">instead of re-using prototype, create a program anyway (ignoring the validator) in case it is the last try. Or even better: Make the validator return a defect rate!  </p>
</dd>
<dt><a class="anchor" id="_todo000150"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype.html#a4327e7a69fcbcfe051e00e46c933ea7a">org.jgap.gp.impl.GPGenotype.fillPopulation</a>  (final int a_num)</dt>
<dd><p class="startdd">care about </p>
<p class="enddd">implement filling up population  </p>
</dd>
<dt><a class="anchor" id="_todo000142"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype.html#af3246cb6a802ae1cde833013c130f6e9">org.jgap.gp.impl.GPGenotype.randomInitialGenotype</a>  (final GPConfiguration a_conf, Class[] a_types, Class[][] a_argTypes, CommandGene[][] a_nodeSets, int[] a_minDepths, int[] a_maxDepths, int a_maxNodes, boolean[] a_fullModeAllowed, boolean a_verboseOutput, IPopulationCreator a_popCreator)</dt>
<dd><p class="startdd">remove unused nodes from configuration </p>
<p class="enddd">optimize access to map  </p>
</dd>
<dt><a class="anchor" id="_todo000153"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_genotype.html#ad264330c888b133296578decc78f265e">org.jgap.gp.impl.GPGenotype.verifyDepthsForNodes</a>  (GPPopulation a_pop, GPConfiguration a_conf, Class[] a_types, int[] a_minDepths, int[] a_maxDepths, int a_maxNodes, CommandGene[][] a_nodeSets)</dt>
<dd><p class="startdd">impl </p>
<p class="enddd">impl  </p>
</dd>
<dt><a class="anchor" id="_todo000155"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population.html#a21670302d6d9c5ad4bdc2903797eabff">org.jgap.gp.impl.GPPopulation.create</a>  (Class[] a_types, Class[][] a_argTypes, CommandGene[][] a_nodeSets, int[] a_minDepths, int[] a_maxDepths, int a_maxNodes, boolean[] a_fullModeAllowed, IProgramCreator a_programCreator, int a_offset)</dt>
<dd><p class="startdd">add element of randomness </p>
<p class="enddd">5: make configurable  </p>
</dd>
<dt><a class="anchor" id="_todo000189"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_test.html#aeeefdb7d33e4e7555cecc9591989b3e1">org.jgap.gp.impl.GPPopulationTest.testCreate_1</a>  ()</dt>
<dd>adapt  </dd>
<dt><a class="anchor" id="_todo000190"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_test.html#a0c2d02f51697a84b44ea7a047a2f838e">org.jgap.gp.impl.GPPopulationTest.testCreate_2</a>  ()</dt>
<dd>adapt  </dd>
<dt><a class="anchor" id="_todo000191"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_g_p_population_test.html#aa41a4305aac81ed18627fda12889df58">org.jgap.gp.impl.GPPopulationTest.testSetPrototype_0</a>  ()</dt>
<dd>assert that prototype is only set automatically if none is preset  </dd>
<dt><a class="anchor" id="_todo000158"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#ac9896e142c28c71711133698041fb72f">org.jgap.gp.impl.ProgramChromosome.compareTo</a>  (Object a_other)</dt>
<dd>compare m_functionSet  </dd>
<dt><a class="anchor" id="_todo000157"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1gp_1_1impl_1_1_program_chromosome.html#a1ca14815335297629dc3bd4812445405">org.jgap.gp.impl.ProgramChromosome.growOrFullNode</a>  (int a_num, int a_depth, Class a_returnType, int a_subReturnType, CommandGene[] a_functionSet, CommandGene a_rootNode, int a_recurseLevel, boolean a_grow, int a_childNum, boolean a_validateNode)</dt>
<dd>we could optionally use the clone handler  </dd>
<dt><a class="anchor" id="_todo000192"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1gui_1_1_config_writer_test.html#a689d2d62adc54d0523e5c94ded843546">org.jgap.gui.ConfigWriterTest.testConfigData_0</a>  ()</dt>
<dd><p class="startdd">write to stream </p>
<p class="enddd">read from stream  </p>
</dd>
<dt><a class="anchor" id="_todo000194"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1impl_1_1_cauchy_random_generator_test.html#a157a2f8e2d9d45bfd8dc63fc63cc4867">org.jgap.impl.CauchyRandomGeneratorTest.testSerialize_0</a>  ()</dt>
<dd>fix test as Java 5 uses java.util.concurrent.AtomicLong instead of sun.misc.AtomicLong  </dd>
<dt><a class="anchor" id="_todo000195"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1impl_1_1_chain_of_selectors_test.html#a2a61fade1cbf2211a2339a2155191839">org.jgap.impl.ChainOfSelectorsTest.testEquals_0</a>  ()</dt>
<dd>improve: 2 unsame selectors of same class with same params should make the chain equal!  </dd>
<dt><a class="anchor" id="_todo000161"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1impl_1_1_crossover_operator.html#a13ba59ced7b3bfd8b7f9433756664726">org.jgap.impl.CrossoverOperator.compareTo</a>  (final Object a_other)</dt>
<dd>consider Configuration  </dd>
<dt><a class="anchor" id="_todo000162"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1impl_1_1_double_gene.html#a9b71158d4e12d38732f9b60bf2ba1949">org.jgap.impl.DoubleGene.setValueFromPersistentRepresentation</a>  (String a_representation)</dt>
<dd>unify first part of method with IntegerGene  </dd>
<dt><a class="anchor" id="_todo000196"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1impl_1_1_double_gene_test.html#a6e78c6ca74eb86d0d457a7c10c05c999">org.jgap.impl.DoubleGeneTest.testSetToRandomValue_2</a>  ()</dt>
<dd>test needed any longer?  </dd>
<dt><a class="anchor" id="_todo000163"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1impl_1_1fitness_1_1_truth_table_fitness_function.html#acd41df30587d5553bf61293c0e025051">org.jgap.impl.fitness.TruthTableFitnessFunction.calcFitness</a>  (final Map a_actualInputOutput)</dt>
<dd>consider length of formula (i.e. number of terms, e.g.) for fitness calculation  </dd>
<dt><a class="anchor" id="_todo000164"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1impl_1_1_fittest_population_merger.html#ae0bd6e67df9b565c419733478078e384">org.jgap.impl.FittestPopulationMerger.mergePopulations</a>  (final Population a_population1, final Population a_population2, final int a_new_population_size)</dt>
<dd>check if configurations of both pops are equal resp. their fitness evaluators!  </dd>
<dt><a class="anchor" id="_todo000197"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1impl_1_1_fixed_binary_gene_test.html#a7664b9a8cb84c4a3cc420ef9aef5b81b">org.jgap.impl.FixedBinaryGeneTest.testHashCode_0</a>  ()</dt>
<dd>implement checks for uniqueness  </dd>
<dt><a class="anchor" id="_todo000167"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1impl_1_1_g_a_breeder.html#ae2fbd377bce110b40fd5f1190f172b7e">org.jgap.impl.GABreeder.bulkFunctionEvaluation</a>  (Configuration a_conf, BulkFitnessFunction a_bulkFunction, Population a_pop, boolean a_monitorActive)</dt>
<dd>utilize jobs: bulk fitness function is not so important for a prototype!  </dd>
<dt><a class="anchor" id="_todo000165"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1impl_1_1_g_a_breeder.html#a2714ceb0e7ce43593588c15dd0c7463e">org.jgap.impl.GABreeder.evolve</a>  (Population a_pop, Configuration a_conf)</dt>
<dd>utilize jobs. In pop do also utilize jobs, especially for fitness computation  </dd>
<dt><a class="anchor" id="_todo000199"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1impl_1_1_gaussian_random_generator_test.html#ace1d14a1c8b90112d955db90b5e22eb8">org.jgap.impl.GaussianRandomGeneratorTest.testGetGaussianStdDeviation_0</a>  ()</dt>
<dd>finish  </dd>
<dt><a class="anchor" id="_todo000200"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1impl_1_1_gaussian_random_generator_test.html#a340a0b166f5094800a2e10b63e258bec">org.jgap.impl.GaussianRandomGeneratorTest.testSerialize_0</a>  ()</dt>
<dd>fix test as Java 5 uses java.util.concurrent.AtomicLong instead of sun.misc.AtomicLong  </dd>
<dt><a class="anchor" id="_todo000201"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1impl_1_1_gaussian_random_generator_test.html#a9710b02b268a1019979c45a065789341">org.jgap.impl.GaussianRandomGeneratorTest.testSerialize_1</a>  ()</dt>
<dd>fix test as Java 5 uses java.util.concurrent.AtomicLong instead of sun.misc.AtomicLong  </dd>
<dt><a class="anchor" id="_todo000168"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1impl_1_1_j_g_a_p_factory.html#a131e58c62f763615b3e47555a30a5fea">org.jgap.impl.JGAPFactory.clone</a>  ()</dt>
<dd>check if it works this way  </dd>
<dt><a class="anchor" id="_todo000171"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1impl_1_1job_1_1_evolve_job.html#a39f8a447fa201c74de218cd06ca0c5bb">org.jgap.impl.job.EvolveJob.applyGeneticOperators</a>  (Configuration a_config, Population a_pop)</dt>
<dd>utilize jobs: integrate job into GeneticOperator  </dd>
<dt><a class="anchor" id="_todo000169"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1impl_1_1job_1_1_evolve_job.html#a128b696a7adc919ae743ad7980b59927">org.jgap.impl.job.EvolveJob.applyNaturalSelectors</a>  (Configuration a_config, Population a_pop, boolean a_processBeforeGeneticOperators)</dt>
<dd><p class="startdd">optionally use working pool </p>
<p class="enddd">utilize jobs: integrate job into NaturalSelector!  </p>
</dd>
<dt><a class="anchor" id="_todo000172"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1impl_1_1job_1_1_job_base.html#a4395288c6b1143da26fc7edb5d2bc792">org.jgap.impl.job.JobBase.run</a>  ()</dt>
<dd>what to do here?  </dd>
<dt><a class="anchor" id="_todo000202"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1impl_1_1_map_gene_test.html#a7c02a53a8930dc6240785629c6c340d7">org.jgap.impl.MapGeneTest.testSetToRandomValue_0</a>  ()</dt>
<dd>add test for applyMutation  </dd>
<dt><a class="anchor" id="_todo000173"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1impl_1_1_mutation_operator.html#a4a1c852770559bba4ffa3e9718f4a0d8">org.jgap.impl.MutationOperator.operate</a>  (final Population a_population, final List a_candidateChromosomes)</dt>
<dd>move to base class, refactor  </dd>
<dt><a class="anchor" id="_todo000203"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1impl_1_1_set_gene_test.html#a41e2d57a098dcfc8358e50f7117efbcf">org.jgap.impl.SetGeneTest.testApplyMutation_3</a>  ()</dt>
<dd>from here on: adapt tests to SetGene  </dd>
<dt><a class="anchor" id="_todo000204"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1impl_1_1_swapping_mutation_operator_test.html#a92fbca72f89b7df34a3bb3d2047f10ee">org.jgap.impl.SwappingMutationOperatorTest.testOperate_1</a>  ()</dt>
<dd>assert result is correct  </dd>
<dt><a class="anchor" id="_todo000174"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1impl_1_1_threshold_selector.html#a0503b950969a0c79b42963391ca131f5">org.jgap.impl.ThresholdSelector.selectChromosomes</a>  (final int a_howManyToSelect, Population a_to_pop)</dt>
<dd>replace this step by adding newly to create chromosomes  </dd>
<dt><a class="anchor" id="_todo000175"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1impl_1_1_two_way_mutation_operator.html#a53ac73cd4e6b197e6756d4efe02bd01e">org.jgap.impl.TwoWayMutationOperator.operate</a>  (final Population a_population, final List a_candidateChromosomes)</dt>
<dd><p class="startdd">make this configurable, it is a first test, see example DynamicMutationExample </p>
<p class="enddd">move to base class, refactor  </p>
</dd>
<dt><a class="anchor" id="_todo000206"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1_j_g_a_p_test_case.html#a3ccc244fd7637efacc27641a3b4bac99">org.jgap.JGAPTestCase.assertEqualsMap</a>  (Map a_map1, Map a_map2)</dt>
<dd>implement  </dd>
<dt><a class="anchor" id="_todo000177"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1_population.html#a7dd9dbd2fecb4e2c4841787fd78ea64c">org.jgap.Population.keepPopSizeConstant</a>  ()</dt>
<dd>use dedicated selector for that  </dd>
<dt><a class="anchor" id="_todo000207"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1supergenes_1_1_supergene_persistent_representation_test.html#a6cae74832bb9a8131bed212047974f89">org.jgap.supergenes.SupergenePersistentRepresentationTest.testReset_0</a>  ()</dt>
<dd>care that m_immutable is filled  </dd>
<dt><a class="anchor" id="_todo000179"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1util_1_1_class_kit.html#a2eb98cd025d7bb9755a162b9392fa5d6">org.jgap.util.ClassKit.find</a>  (final String a_tosubclassname)</dt>
<dd>take care of abstract classes &ndash;&gt; introduce parameter for that  </dd>
<dt><a class="anchor" id="_todo000180"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1util_1_1_class_kit.html#ab3cdf47ea2ab88910b62eb16425585b2">org.jgap.util.ClassKit.getPlugins</a>  (final String a_directory)</dt>
<dd><p class="startdd">add input param: type (or list of types) to look for </p>
<p class="enddd">check if class assignable from given type  </p>
</dd>
<dt><a class="anchor" id="_todo000183"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1util_1_1_file_kit.html#af68540ad9faa606310ee8952ca57d894">org.jgap.util.FileKit.getFilename</a>  (String name_and_path, String fileseparator)</dt>
<dd>possibly use File.getName() instead?  </dd>
<dt><a class="anchor" id="_todo000209"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1util_1_1_number_kit_test.html#a1b68aa7e58cef0016f48dc2208b4db38">org.jgap.util.NumberKitTest.testNiceDecimalNumber_0</a>  ()</dt>
<dd>this is not an optimal result  </dd>
<dt><a class="anchor" id="_todo000210"></a>Miembro <a class="el" href="classorg_1_1jgap_1_1util_1_1_plugin_discoverer_test.html#a7c18c34fa61eea040b9d90bb21d8ada5">org.jgap.util.PluginDiscovererTest.testConstruct_0</a>  ()</dt>
<dd>implement </dd>
</dl>
</div></div><!-- contents -->
</div><!-- doc-content -->
<!-- start footer part -->
<div id="nav-path" class="navpath"><!-- id is needed for treeview function! -->
  <ul>
    <li class="footer">Generado el Domingo, 24 de Noviembre de 2013 18:22:47 para Proyecto JGAP por
    <a href="http://www.doxygen.org/index.html">
    <img class="footer" src="doxygen.png" alt="doxygen"/></a> 1.8.5 </li>
  </ul>
</div>
</body>
</html>
